// busboy-express/dist/src/index.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
var Busboy = require("busboy");
var Debug = require("debug");
var fs = require("fs-extra");
var os = require("os");
var path = require("path");
var uniqid = require("uniqid");
var util = require("util");
var debug = Debug('busboy-express');
var defaultOptions = {
    multipartOnly: true,
    uploadPath: os.tmpdir(),
    fields: true,
    files: [],
};
function busboyExpress(mwOptions) {
    var options = Object.assign({}, defaultOptions, mwOptions || {});
    return function busboyMiddleware(req, res, next) {
        var _this = this;
        try {
            if (req.method !== 'POST') {
                next();
            }
            if (options.multipartOnly &&
                (!req.headers['content-type'] ||
                    !req.headers['content-type'].includes('multipart/form-data'))) {
                next();
            }
            if ((!options.fields || (options.fields instanceof Array && !options.fields.length)) &&
                (!options.files || (options.files instanceof Array && !options.files.length))) {
                // Nothing to do.
                next();
            }
            debug(req.method + " " + req.url);
            req.files = {};
            req.fields = {};
            var fileWritePromises_1 = [];
            var busboy = new Busboy({ headers: req.headers });
            busboy.on('file', function (fieldname, stream, filename, encoding, mimetype) {
                if (!options.files ||
                    (options.files instanceof Array && !options.files.includes(fieldname))) {
                    stream.resume();
                    return;
                }
                var targetPath = path.join(options.uploadPath || defaultOptions.uploadPath, uniqid());
                debug(req.method + " " + req.url +
                    (" FILE [" + fieldname + "]:") +
                    (" filename: " + filename + ", ") +
                    (" encoding: " + encoding + ", ") +
                    (" mimetype: " + mimetype + ", ") +
                    (" targetPath: " + targetPath) +
                    "");
                fileWritePromises_1.push(new Promise(function (resolve, reject) {
                    try {
                        stream.on('error', function (err) {
                            debug(req.method + " " + req.url +
                                " READ ERROR: " +
                                (" FILE [" + fieldname + "]:") +
                                (" filename: " + filename + " ") +
                                "", err);
                            resolve();
                        });
                        var wStream_1 = fs.createWriteStream(targetPath);
                        wStream_1.on('error', function (err) {
                            debug(req.method + " " + req.url +
                                " WRITE ERROR: " +
                                (" FILE [" + fieldname + "]:") +
                                (" filename: " + filename + " ") +
                                "", err);
                        });
                        stream.pipe(wStream_1, { end: false });
                        stream.on('end', function () {
                            wStream_1.end(function () { return __awaiter(_this, void 0, void 0, function () {
                                var stats, err_1;
                                return __generator(this, function (_a) {
                                    switch (_a.label) {
                                        case 0:
                                            _a.trys.push([0, 2, , 3]);
                                            return [4 /*yield*/, fs.stat(targetPath)];
                                        case 1:
                                            stats = _a.sent();
                                            if (!req.files.hasOwnProperty(fieldname)) {
                                                req.files[fieldname] = [];
                                            }
                                            req.files[fieldname].push({
                                                fieldname: fieldname,
                                                path: targetPath,
                                                size: stats.size,
                                                filename: filename,
                                                encoding: encoding,
                                                mimetype: mimetype,
                                            });
                                            resolve();
                                            return [3 /*break*/, 3];
                                        case 2:
                                            err_1 = _a.sent();
                                            debug(req.method + " " + req.url +
                                                " STAT ERROR: " +
                                                (" FILE [" + fieldname + "]:") +
                                                (" filename: " + filename + " ") +
                                                "", err_1);
                                            resolve();
                                            return [3 /*break*/, 3];
                                        case 3: return [2 /*return*/];
                                    }
                                });
                            }); });
                        });
                    }
                    catch (err) {
                        debug(req.method + " " + req.url +
                            " FILE ERROR: " +
                            (" FILE [" + fieldname + "]:") +
                            (" filename: " + filename + " ") +
                            "", err);
                    }
                }));
            });
            busboy.on('field', function (fieldname, val, fieldnameTruncated, valTruncated, encoding, mimetype) {
                if (!options.fields ||
                    (options.fields instanceof Array && !options.fields.includes(fieldname))) {
                    return;
                }
                debug(req.method + " " + req.url +
                    (" FIELD [" + fieldname + "]:") +
                    (" value: " + util.inspect(val) + ", ") +
                    (" encoding: " + encoding + ", ") +
                    (" mimetype: " + mimetype + ", ") +
                    "");
                if (!req.fields.hasOwnProperty(fieldname)) {
                    req.fields[fieldname] = [];
                }
                req.fields[fieldname].push(val);
            });
            busboy.on('finish', function () { return __awaiter(_this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            debug(req.method + " " + req.url + " FINISH, awaiting file writes");
                            return [4 /*yield*/, Promise.all(fileWritePromises_1)];
                        case 1:
                            _a.sent();
                            debug(req.method + " " + req.url + " DONE");
                            next();
                            return [2 /*return*/];
                    }
                });
            }); });
            busboy.on('error', function (err) {
                debug(req.method + " " + req.url + " ERROR", err);
                next(err);
            });
            req.pipe(busboy);
        }
        catch (err) {
            next(err);
        }
    };
}
exports.busboyExpress = busboyExpress;
exports.busboy = busboyExpress;
function busboyExpressCleanup(req) {
    return __awaiter(this, void 0, void 0, function () {
        var err_2;
        var _this = this;
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0:
                    if (!req.files) return [3 /*break*/, 4];
                    _a.label = 1;
                case 1:
                    _a.trys.push([1, 3, , 4]);
                    return [4 /*yield*/, Promise.all(Object.keys(req.files).map(function (fieldname) { return __awaiter(_this, void 0, void 0, function () {
                            var _this = this;
                            return __generator(this, function (_a) {
                                switch (_a.label) {
                                    case 0: return [4 /*yield*/, Promise.all(req.files[fieldname].map(function (file) { return __awaiter(_this, void 0, void 0, function () {
                                            var err_3;
                                            return __generator(this, function (_a) {
                                                switch (_a.label) {
                                                    case 0:
                                                        _a.trys.push([0, 4, , 5]);
                                                        return [4 /*yield*/, fs.pathExists(file.path)];
                                                    case 1:
                                                        if (!_a.sent()) return [3 /*break*/, 3];
                                                        return [4 /*yield*/, fs.remove(file.path)];
                                                    case 2:
                                                        _a.sent();
                                                        _a.label = 3;
                                                    case 3: return [3 /*break*/, 5];
                                                    case 4:
                                                        err_3 = _a.sent();
                                                        debug("Failed to cleanup file: " + file.path + " (" + file.filename, err_3);
                                                        return [3 /*break*/, 5];
                                                    case 5: return [2 /*return*/];
                                                }
                                            });
                                        }); }))];
                                    case 1:
                                        _a.sent();
                                        return [2 /*return*/];
                                }
                            });
                        }); }))];
                case 2:
                    _a.sent();
                    return [3 /*break*/, 4];
                case 3:
                    err_2 = _a.sent();
                    debug("Error cleaning up uploaded files", err_2);
                    return [3 /*break*/, 4];
                case 4: return [2 /*return*/];
            }
        });
    });
}
exports.cleanup = busboyExpressCleanup;
busboyExpress.cleanup = busboyExpressCleanup;
exports.default = busboyExpress;
