// @linode/api-v4/lib/index.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));

// src/account/account.ts
import {
  updateAccountSchema,
  UpdateAccountSettingsSchema
} from "@linode/validation/lib/account.schema";

// src/constants.ts
var API_ROOT = "https://api.linode.com/v4";
var BETA_API_ROOT = API_ROOT + "beta";

// src/request.ts
import Axios, {
  AxiosHeaders
} from "axios";
var baseRequest = Axios.create({
  baseURL: "https://api.linode.com/v4"
});
baseRequest.interceptors.request.use((config) => {
  const isRunningInNode = typeof process === "object";
  if (!isRunningInNode) {
    return config;
  }
  const headers = new AxiosHeaders(config.headers);
  headers.set("User-Agent", "linodejs");
  return __spreadProps(__spreadValues({}, config), { headers });
});
var setToken = (token) => {
  return baseRequest.interceptors.request.use((config) => {
    const headers = new AxiosHeaders(config.headers);
    headers.set("Authorization", `Bearer ${token}`);
    return __spreadProps(__spreadValues({}, config), { headers });
  });
};
var set = (field, value) => (object) => {
  return !isEmpty(value) ? __spreadProps(__spreadValues({}, object), { [field]: value }) : object;
};
var isEmpty = (v) => v === void 0 || v === null || v.length === 0 || typeof v === "object" && Object.keys(v).length === 0 && v.constructor === Object;
var setURL = (url) => set("url", url);
var setMethod = (method) => set("method", method);
var setParams = (params) => set("params", params);
var setHeaders = (newHeaders = {}) => (object) => {
  return !isEmpty(newHeaders) ? __spreadProps(__spreadValues({}, object), { headers: __spreadValues(__spreadValues({}, object.headers), newHeaders) }) : object;
};
var setData = (data, schema, postValidationTransform) => {
  if (!schema) {
    return set("data", data);
  }
  const updatedData = typeof postValidationTransform === "function" ? postValidationTransform(data) : data;
  try {
    schema.validateSync(data, { abortEarly: false });
    return set("data", updatedData);
  } catch (error) {
    return (object) => __spreadProps(__spreadValues({}, object), {
      data: updatedData,
      validationErrors: convertYupToLinodeErrors(error)
    });
  }
};
var convertYupToLinodeErrors = (validationError) => {
  const { inner } = validationError;
  if (inner && inner.length > 0) {
    return inner.reduce((result, innerValidationError) => {
      const err = convertYupToLinodeErrors(innerValidationError);
      return Array.isArray(err) ? [...result, ...err] : [...result, err];
    }, []);
  }
  return [mapYupToLinodeAPIError(validationError)];
};
var mapYupToLinodeAPIError = ({
  message,
  path
}) => __spreadValues({
  reason: message
}, path && { field: path });
var setXFilter = (xFilter) => {
  return (object) => !isEmpty(xFilter) ? __spreadProps(__spreadValues({}, object), {
    headers: __spreadProps(__spreadValues({}, object.headers), { "X-Filter": JSON.stringify(xFilter) })
  }) : object;
};
var reduceRequestConfig = (...fns) => fns.reduceRight((result, fn) => fn(result), {
  url: "https://api.linode.com/v4",
  headers: {}
});
var requestGenerator = (...fns) => {
  const config = reduceRequestConfig(...fns);
  if (config.validationErrors) {
    return Promise.reject(
      config.validationErrors
      // All failed requests, client or server errors, should be APIError[]
    );
  }
  return baseRequest(config).then((response) => response.data);
};
var request_default = requestGenerator;

// src/account/account.ts
var getAccountInfo = () => {
  return request_default(setURL(`${API_ROOT}/account`), setMethod("GET"));
};
var getNetworkUtilization = () => request_default(
  setURL(`${API_ROOT}/account/transfer`),
  setMethod("GET")
);
var updateAccountInfo = (data) => request_default(
  setURL(`${API_ROOT}/account`),
  setMethod("PUT"),
  setData(data, updateAccountSchema)
);
var getAccountSettings = () => request_default(
  setURL(`${API_ROOT}/account/settings`),
  setMethod("GET")
);
var updateAccountSettings = (data) => request_default(
  setURL(`${API_ROOT}/account/settings`),
  setMethod("PUT"),
  setData(data, UpdateAccountSettingsSchema)
);
var cancelAccount = (data) => {
  return request_default(
    setURL(`${API_ROOT}/account/cancel`),
    setMethod("POST"),
    setData(data)
  );
};
var getAccountAgreements = () => request_default(
  setURL(`${BETA_API_ROOT}/account/agreements`),
  setMethod("GET")
);
var getAccountAvailabilities = (params, filter) => request_default(
  setURL(`${BETA_API_ROOT}/account/availability`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getAccountAvailability = (regionId) => request_default(
  setURL(
    `${BETA_API_ROOT}/account/availability/${encodeURIComponent(regionId)}`
  ),
  setMethod("GET")
);
var signAgreement = (data) => {
  return request_default(
    setURL(`${BETA_API_ROOT}/account/agreements`),
    setMethod("POST"),
    setData(data)
  );
};
var getChildAccounts = ({ filter, params, headers }) => request_default(
  setURL(`${API_ROOT}/account/child-accounts`),
  setMethod("GET"),
  setHeaders(headers),
  setParams(params),
  setXFilter(filter)
);
var getChildAccount = ({ euuid, headers }) => request_default(
  setURL(`${API_ROOT}/account/child-accounts/${encodeURIComponent(euuid)}`),
  setMethod("GET"),
  setHeaders(headers)
);
var createChildAccountPersonalAccessToken = ({
  euuid,
  headers
}) => request_default(
  setURL(
    `${API_ROOT}/account/child-accounts/${encodeURIComponent(euuid)}/token`
  ),
  setMethod("POST"),
  setHeaders(headers),
  setData(euuid)
);

// src/account/betas.ts
var getAccountBetas = (params, filter) => request_default(
  setURL(`${API_ROOT}/account/betas`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getAccountBeta = (betaId) => request_default(
  setURL(`${API_ROOT}/account/betas/${encodeURIComponent(betaId)}`),
  setMethod("GET")
);
var enrollInBeta = (data) => request_default(
  setURL(`${API_ROOT}/account/betas`),
  setMethod("POST"),
  setData(data)
);

// src/account/events.ts
var getEvents = (params = {}, filter = {}) => request_default(
  setURL(`${API_ROOT}/account/events`),
  setMethod("GET"),
  setXFilter(filter),
  setParams(params)
);
var getEvent = (eventId) => request_default(
  setURL(`${API_ROOT}/account/events/${encodeURIComponent(eventId)}`),
  setMethod("GET")
);
var markEventSeen = (eventId) => request_default(
  setURL(`${API_ROOT}/account/events/${encodeURIComponent(eventId)}/seen`),
  setMethod("POST")
);
var markEventRead = (eventId) => request_default(
  setURL(`${API_ROOT}/account/events/${encodeURIComponent(eventId)}/read`),
  setMethod("POST")
);
var getNotifications = (params, filter) => request_default(
  setURL(`${API_ROOT}/account/notifications`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);

// src/account/invoices.ts
var getInvoices = (params, filter) => request_default(
  setURL(`${API_ROOT}/account/invoices`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getInvoice = (invoiceId) => request_default(
  setURL(`${API_ROOT}/account/invoices/${encodeURIComponent(invoiceId)}`),
  setMethod("GET")
);
var getInvoiceItems = (invoiceId, params, filter) => request_default(
  setURL(
    `${API_ROOT}/account/invoices/${encodeURIComponent(invoiceId)}/items`
  ),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);

// src/account/payments.ts
import {
  CreditCardSchema,
  PaymentSchema,
  PaymentMethodSchema
} from "@linode/validation/lib/account.schema";
var getPayments = (params, filter) => request_default(
  setURL(`${API_ROOT}/account/payments`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var makePayment = (data) => {
  if (!data.cvv) {
    delete data.cvv;
  }
  return request_default(
    setURL(`${API_ROOT}/account/payments`),
    setMethod("POST"),
    setData(data, PaymentSchema)
  );
};
var saveCreditCard = (data) => {
  return request_default(
    setURL(`${API_ROOT}/account/credit-card`),
    setMethod("POST"),
    setData(data, CreditCardSchema)
  );
};
var getPaymentMethods = (params) => {
  return request_default(
    setURL(`${API_ROOT}/account/payment-methods`),
    setMethod("GET"),
    setParams(params)
  );
};
var getPaymentMethod = (id) => {
  return request_default(
    setURL(`${API_ROOT}/account/payment-method/${encodeURIComponent(id)}`),
    setMethod("GET")
  );
};
var getClientToken = () => {
  return request_default(
    setURL(`${API_ROOT}/account/client-token`),
    setMethod("GET")
  );
};
var addPaymentMethod = (data) => {
  return request_default(
    setURL(`${API_ROOT}/account/payment-methods`),
    setMethod("POST"),
    setData(data, PaymentMethodSchema)
  );
};
var makeDefaultPaymentMethod = (id) => {
  return request_default(
    setURL(
      `${API_ROOT}/account/payment-methods/${encodeURIComponent(
        id
      )}/make-default`
    ),
    setMethod("POST")
  );
};
var deletePaymentMethod = (id) => {
  return request_default(
    setURL(`${API_ROOT}/account/payment-methods/${encodeURIComponent(id)}`),
    setMethod("DELETE")
  );
};

// src/account/users.ts
import {
  CreateUserSchema,
  UpdateUserSchema
} from "@linode/validation/lib/account.schema";
var getUsers = (params, filters) => request_default(
  setURL(`${API_ROOT}/account/users`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filters)
);
var getUser = (username) => request_default(
  setURL(`${API_ROOT}/account/users/${encodeURIComponent(username)}`),
  setMethod("GET")
);
var createUser = (data) => request_default(
  setURL(`${API_ROOT}/account/users`),
  setMethod("POST"),
  setData(data, CreateUserSchema)
);
var updateUser = (username, data) => request_default(
  setURL(`${API_ROOT}/account/users/${encodeURIComponent(username)}`),
  setMethod("PUT"),
  setData(data, UpdateUserSchema)
);
var deleteUser = (username) => request_default(
  setURL(`${API_ROOT}/account/users/${encodeURIComponent(username)}`),
  setMethod("DELETE")
);
var getGrants = (username) => request_default(
  setURL(`${API_ROOT}/account/users/${encodeURIComponent(username)}/grants`),
  setMethod("GET")
);
var updateGrants = (username, data) => request_default(
  setURL(`${API_ROOT}/account/users/${encodeURIComponent(username)}/grants`),
  setMethod("PUT"),
  setData(data)
);

// src/account/oauth.ts
import {
  createOAuthClientSchema,
  updateOAuthClientSchema
} from "@linode/validation/lib/account.schema";
var getOAuthClients = (params, filter) => request_default(
  setURL(`${API_ROOT}/account/oauth-clients`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getOAuthClient = (clientId) => request_default(
  setURL(`${API_ROOT}/account/oauth-clients/${encodeURIComponent(clientId)}`),
  setMethod("GET")
);
var createOAuthClient = (data) => request_default(
  setURL(`${API_ROOT}/account/oauth-clients`),
  setMethod("POST"),
  setData(data, createOAuthClientSchema)
);
var resetOAuthClientSecret = (clientId) => request_default(
  setURL(
    `${API_ROOT}/account/oauth-clients/${encodeURIComponent(
      clientId
    )}/reset-secret`
  ),
  setMethod("POST")
);
var updateOAuthClient = (clientId, data) => request_default(
  setURL(`${API_ROOT}/account/oauth-clients/${encodeURIComponent(clientId)}`),
  setMethod("PUT"),
  setData(data, updateOAuthClientSchema)
);
var deleteOAuthClient = (clientId) => request_default(
  setURL(`${API_ROOT}/account/oauth-clients/${encodeURIComponent(clientId)}`),
  setMethod("DELETE")
);

// src/account/promos.ts
import { PromoCodeSchema } from "@linode/validation/lib/account.schema";
var addPromotion = (code) => request_default(
  setURL(`${API_ROOT}/account/promo-codes`),
  setMethod("POST"),
  setData({ promo_code: code }, PromoCodeSchema)
);

// src/account/maintenance.ts
var getAccountMaintenance = (params, filter) => request_default(
  setURL(`${API_ROOT}/account/maintenance`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);

// src/account/logins.ts
var getAccountLogins = (params, filter) => request_default(
  setURL(`${API_ROOT}/account/logins`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getAccountLogin = (loginId) => request_default(
  setURL(`${API_ROOT}/account/logins/${encodeURIComponent(loginId)}`),
  setMethod("GET")
);

// src/aclb/configurations.ts
import {
  CreateConfigurationSchema,
  UpdateConfigurationSchema
} from "@linode/validation";
var getLoadbalancerConfigurations = (loadbalancerId, params, filter) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/configurations`
  ),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getLoadbalancerConfiguration = (loadbalancerId, configurationId) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/configurations/${encodeURIComponent(configurationId)}`
  ),
  setMethod("GET")
);
var getLoadbalancerConfigurationsEndpointHealth = (loadbalancerId) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/configurations/endpoints-health`
  ),
  setMethod("GET")
);
var createLoadbalancerConfiguration = (loadbalancerId, data) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/configurations`
  ),
  setData(data, CreateConfigurationSchema),
  setMethod("POST")
);
var updateLoadbalancerConfiguration = (loadbalancerId, configurationId, data) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/configurations/${encodeURIComponent(configurationId)}`
  ),
  setData(data, UpdateConfigurationSchema),
  setMethod("PUT")
);
var deleteLoadbalancerConfiguration = (loadbalancerId, configurationId) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/configurations/${encodeURIComponent(configurationId)}`
  ),
  setMethod("DELETE")
);

// src/aclb/loadbalancers.ts
import { CreateBasicLoadbalancerSchema } from "@linode/validation";
var getLoadbalancers = (params, filter) => request_default(
  setURL(`${BETA_API_ROOT}/aclb`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getLoadbalancer = (id) => request_default(
  setURL(`${BETA_API_ROOT}/aclb/${encodeURIComponent(id)}`),
  setMethod("GET")
);
var getLoadbalancerEndpointHealth = (id) => request_default(
  setURL(`${BETA_API_ROOT}/aclb/${encodeURIComponent(id)}/endpoints-health`),
  setMethod("GET")
);
var createLoadbalancer = (data) => request_default(
  setURL(`${BETA_API_ROOT}/aclb`),
  setData(data),
  setMethod("POST")
);
var createBasicLoadbalancer = (data) => request_default(
  setURL(`${BETA_API_ROOT}/aclb`),
  setData(data, CreateBasicLoadbalancerSchema),
  setMethod("POST")
);
var updateLoadbalancer = (id, data) => request_default(
  setURL(`${BETA_API_ROOT}/aclb/${encodeURIComponent(id)}`),
  setData(data),
  setMethod("PUT")
);
var deleteLoadbalancer = (id) => request_default(
  setURL(`${BETA_API_ROOT}/aclb/${encodeURIComponent(id)}`),
  setMethod("DELETE")
);

// src/aclb/routes.ts
import { UpdateRouteSchema, CreateRouteSchema } from "@linode/validation";
var getLoadbalancerRoutes = (loadbalancerId, params, filter) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(loadbalancerId)}/routes`
  ),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getLoadbalancerRoute = (loadbalancerId, routeId) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/routes/${encodeURIComponent(routeId)}`
  ),
  setMethod("GET")
);
var createLoadbalancerRoute = (loadbalancerId, data) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(loadbalancerId)}/routes`
  ),
  setData(data, CreateRouteSchema),
  setMethod("POST")
);
var updateLoadbalancerRoute = (loadbalancerId, routeId, data) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/routes/${encodeURIComponent(routeId)}`
  ),
  setData(data, UpdateRouteSchema),
  setMethod("PUT")
);
var deleteLoadbalancerRoute = (loadbalancerId, routeId) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/routes/${encodeURIComponent(routeId)}`
  ),
  setMethod("DELETE")
);

// src/aclb/service-targets.ts
import {
  CreateServiceTargetSchema,
  UpdateServiceTargetSchema
} from "@linode/validation";
var getLoadbalancerServiceTargets = (loadbalancerId, params, filter) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/service-targets`
  ),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getServiceTarget = (loadbalancerId, serviceTargetId) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/service-targets/${encodeURIComponent(serviceTargetId)}`
  ),
  setMethod("GET")
);
var getServiceTargetsEndpointHealth = (loadbalancerId) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/service-targets/endpoints-health`
  ),
  setMethod("GET")
);
var createLoadbalancerServiceTarget = (loadbalancerId, data) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/service-targets`
  ),
  setData(data, CreateServiceTargetSchema),
  setMethod("POST")
);
var updateLoadbalancerServiceTarget = (loadbalancerId, serviceTargetId, data) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/service-targets/${encodeURIComponent(serviceTargetId)}`
  ),
  setData(data, UpdateServiceTargetSchema),
  setMethod("PUT")
);
var deleteLoadbalancerServiceTarget = (loadbalancerId, serviceTargetId) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/service-targets/${encodeURIComponent(serviceTargetId)}`
  ),
  setMethod("DELETE")
);

// src/aclb/certificates.ts
import {
  CreateCertificateSchema,
  UpdateCertificateSchema
} from "@linode/validation";
var getLoadbalancerCertificates = (loadbalancerId, params, filter) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(loadbalancerId)}/certificates`
  ),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getLoadbalancerCertificate = (loadbalancerId, certificateId) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/certificates/${encodeURIComponent(certificateId)}`
  ),
  setMethod("GET")
);
var createLoadbalancerCertificate = (loadbalancerId, data) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(loadbalancerId)}/certificates`
  ),
  setMethod("POST"),
  setData(data, CreateCertificateSchema)
);
var updateLoadbalancerCertificate = (loadbalancerId, certificateId, data) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/certificates/${encodeURIComponent(certificateId)}`
  ),
  setMethod("PUT"),
  setData(data, UpdateCertificateSchema)
);
var deleteLoadbalancerCertificate = (loadbalancerId, certificateId) => request_default(
  setURL(
    `${BETA_API_ROOT}/aclb/${encodeURIComponent(
      loadbalancerId
    )}/certificates/${encodeURIComponent(certificateId)}`
  ),
  setMethod("DELETE")
);

// src/databases/databases.ts
import {
  createDatabaseSchema,
  updateDatabaseSchema
} from "@linode/validation/lib/databases.schema";
var getDatabases = (params, filter) => request_default(
  setURL(`${BETA_API_ROOT}/databases/instances`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getDatabaseTypes = (params, filter) => request_default(
  setURL(`${BETA_API_ROOT}/databases/types`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getDatabaseType = (typeSlug) => request_default(
  setURL(`${BETA_API_ROOT}/databases/types/${encodeURIComponent(typeSlug)}`),
  setMethod("GET")
);
var getDatabaseEngines = (params, filter) => request_default(
  setURL(`${BETA_API_ROOT}/databases/engines`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getDatabaseEngine = (engineSlug) => request_default(
  setURL(`${BETA_API_ROOT}/databases/engines/${encodeURIComponent(engineSlug)}`),
  setMethod("GET")
);
var createDatabase = (engine = "mysql", data) => request_default(
  setURL(`${BETA_API_ROOT}/databases/${encodeURIComponent(engine)}/instances`),
  setMethod("POST"),
  setData(data, createDatabaseSchema)
);
var getEngineDatabases = (engine, params, filter) => request_default(
  setURL(`${BETA_API_ROOT}/databases/${encodeURIComponent(engine)}/instances`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getEngineDatabase = (engine, databaseID) => request_default(
  setURL(
    `${BETA_API_ROOT}/databases/${encodeURIComponent(
      engine
    )}/instances/${encodeURIComponent(databaseID)}`
  ),
  setMethod("GET")
);
var updateDatabase = (engine, databaseID, data) => request_default(
  setURL(
    `${BETA_API_ROOT}/databases/${encodeURIComponent(
      engine
    )}/instances/${encodeURIComponent(databaseID)}`
  ),
  setMethod("PUT"),
  setData(data, updateDatabaseSchema)
);
var deleteDatabase = (engine, databaseID) => request_default(
  setURL(
    `${BETA_API_ROOT}/databases/${encodeURIComponent(
      engine
    )}/instances/${encodeURIComponent(databaseID)}`
  ),
  setMethod("DELETE")
);
var getDatabaseBackups = (engine, databaseID, params, filter) => request_default(
  setURL(
    `${BETA_API_ROOT}/databases/${encodeURIComponent(
      engine
    )}/instances/${encodeURIComponent(databaseID)}/backups`
  ),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getDatabaseBackup = (engine, databaseID, backupID) => request_default(
  setURL(
    `${BETA_API_ROOT}/databases/${encodeURIComponent(
      engine
    )}/instances/${encodeURIComponent(
      databaseID
    )}/backups/${encodeURIComponent(backupID)}`
  ),
  setMethod("GET")
);
var restoreWithBackup = (engine, databaseID, backupID) => request_default(
  setURL(
    `${BETA_API_ROOT}/databases/${encodeURIComponent(
      engine
    )}/instances/${encodeURIComponent(
      databaseID
    )}/backups/${encodeURIComponent(backupID)}/restore`
  ),
  setMethod("POST")
);
var getDatabaseCredentials = (engine, databaseID) => request_default(
  setURL(
    `${BETA_API_ROOT}/databases/${encodeURIComponent(
      engine
    )}/instances/${encodeURIComponent(databaseID)}/credentials`
  ),
  setMethod("GET")
);
var resetDatabaseCredentials = (engine, databaseID) => request_default(
  setURL(
    `${BETA_API_ROOT}/databases/${encodeURIComponent(
      engine
    )}/instances/${encodeURIComponent(databaseID)}/credentials/reset`
  ),
  setMethod("POST")
);
var getSSLFields = (engine, databaseID) => request_default(
  setURL(
    `${BETA_API_ROOT}/databases/${encodeURIComponent(
      engine
    )}/instances/${encodeURIComponent(databaseID)}/ssl`
  ),
  setMethod("GET")
);

// src/domains/domains.ts
import {
  createDomainSchema,
  importZoneSchema,
  updateDomainSchema
} from "@linode/validation/lib/domains.schema";
var getDomains = (params, filter) => request_default(
  setURL(`${API_ROOT}/domains`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getDomain = (domainId) => request_default(
  setURL(`${API_ROOT}/domains/${encodeURIComponent(domainId)}`),
  setMethod("GET")
);
var createDomain = (data) => request_default(
  setData(data, createDomainSchema),
  setURL(`${API_ROOT}/domains`),
  setMethod("POST")
);
var updateDomain = (domainId, data) => request_default(
  setURL(`${API_ROOT}/domains/${encodeURIComponent(domainId)}`),
  setMethod("PUT"),
  setData(data, updateDomainSchema)
);
var deleteDomain = (domainId) => request_default(
  setURL(`${API_ROOT}/domains/${encodeURIComponent(domainId)}`),
  setMethod("DELETE")
);
var cloneDomain = (domainId, data) => request_default(
  setData(data),
  setURL(`${API_ROOT}/domains/${encodeURIComponent(domainId)}/clone`),
  setMethod("POST")
);
var importZone = (data) => request_default(
  setData(data, importZoneSchema),
  setURL(`${API_ROOT}/domains/import`),
  setMethod("POST")
);
var getDNSZoneFile = (domainId) => request_default(
  setURL(`${API_ROOT}/domains/${encodeURIComponent(domainId)}/zone-file`),
  setMethod("GET")
);

// src/domains/records.ts
import {
  createRecordSchema,
  updateRecordSchema
} from "@linode/validation/lib/records.schema";
var getDomainRecords = (domainId, params) => request_default(
  setURL(`${API_ROOT}/domains/${encodeURIComponent(domainId)}/records`),
  setParams(params),
  setMethod("GET")
);
var getDomainRecord = (domainId, recordId) => request_default(
  setURL(
    `${API_ROOT}/domains/${encodeURIComponent(
      domainId
    )}/records/${encodeURIComponent(recordId)}`
  ),
  setMethod("GET")
);
var createDomainRecord = (domainId, data) => request_default(
  setURL(`${API_ROOT}/domains/${encodeURIComponent(domainId)}/records`),
  setMethod("POST"),
  setData(data, createRecordSchema)
);
var updateDomainRecord = (domainId, recordId, data) => request_default(
  setURL(
    `${API_ROOT}/domains/${encodeURIComponent(
      domainId
    )}/records/${encodeURIComponent(recordId)}`
  ),
  setMethod("PUT"),
  setData(data, updateRecordSchema)
);
var deleteDomainRecord = (domainId, recordId) => request_default(
  setURL(
    `${API_ROOT}/domains/${encodeURIComponent(
      domainId
    )}/records/${encodeURIComponent(recordId)}`
  ),
  setMethod("DELETE")
);

// src/entity-transfers/transfers.ts
import { CreateTransferSchema } from "@linode/validation/lib/transfers.schema";
var getEntityTransfers = (params, filter) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filter),
  setURL(`${BETA_API_ROOT}/account/entity-transfers`)
);
var getEntityTransfer = (token) => request_default(
  setMethod("GET"),
  setURL(
    `${BETA_API_ROOT}/account/entity-transfers/${encodeURIComponent(token)}`
  )
);
var createEntityTransfer = (data) => request_default(
  setMethod("POST"),
  setData(data, CreateTransferSchema),
  setURL(`${BETA_API_ROOT}/account/entity-transfers`)
);
var acceptEntityTransfer = (token) => request_default(
  setMethod("POST"),
  setURL(
    `${BETA_API_ROOT}/account/entity-transfers/${encodeURIComponent(
      token
    )}/accept`
  )
);
var cancelTransfer = (token) => request_default(
  setMethod("DELETE"),
  setURL(
    `${BETA_API_ROOT}/account/entity-transfers/${encodeURIComponent(token)}`
  )
);

// src/firewalls/firewalls.ts
import {
  CreateFirewallSchema,
  FirewallDeviceSchema,
  UpdateFirewallSchema
} from "@linode/validation/lib/firewalls.schema";
var getFirewalls = (params, filter) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filter),
  setURL(`${BETA_API_ROOT}/networking/firewalls`)
);
var getFirewall = (firewallID) => request_default(
  setMethod("GET"),
  setURL(
    `${BETA_API_ROOT}/networking/firewalls/${encodeURIComponent(firewallID)}`
  )
);
var createFirewall = (data) => request_default(
  setMethod("POST"),
  setData(data, CreateFirewallSchema),
  setURL(`${BETA_API_ROOT}/networking/firewalls`)
);
var updateFirewall = (firewallID, data) => request_default(
  setMethod("PUT"),
  setData(data, UpdateFirewallSchema),
  setURL(
    `${BETA_API_ROOT}/networking/firewalls/${encodeURIComponent(firewallID)}`
  )
);
var enableFirewall = (firewallID) => updateFirewall(firewallID, { status: "enabled" });
var disableFirewall = (firewallID) => updateFirewall(firewallID, { status: "disabled" });
var deleteFirewall = (firewallID) => request_default(
  setMethod("DELETE"),
  setURL(
    `${BETA_API_ROOT}/networking/firewalls/${encodeURIComponent(firewallID)}`
  )
);
var getFirewallRules = (firewallID, params, filter) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filter),
  setURL(
    `${BETA_API_ROOT}/networking/firewalls/${encodeURIComponent(
      firewallID
    )}/rules`
  )
);
var updateFirewallRules = (firewallID, data) => request_default(
  setMethod("PUT"),
  setData(data),
  // Validation is too complicated for these; leave it to the API.
  setURL(
    `${BETA_API_ROOT}/networking/firewalls/${encodeURIComponent(
      firewallID
    )}/rules`
  )
);
var getFirewallDevices = (firewallID, params, filter) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filter),
  setURL(
    `${BETA_API_ROOT}/networking/firewalls/${encodeURIComponent(
      firewallID
    )}/devices`
  )
);
var getFirewallDevice = (firewallID, deviceID) => request_default(
  setMethod("GET"),
  setURL(
    `${BETA_API_ROOT}/networking/firewalls/${encodeURIComponent(
      firewallID
    )}/devices/${encodeURIComponent(deviceID)}`
  )
);
var addFirewallDevice = (firewallID, data) => request_default(
  setMethod("POST"),
  setURL(
    `${BETA_API_ROOT}/networking/firewalls/${encodeURIComponent(
      firewallID
    )}/devices`
  ),
  setData(data, FirewallDeviceSchema)
);
var deleteFirewallDevice = (firewallID, deviceID) => request_default(
  setMethod("DELETE"),
  setURL(
    `${BETA_API_ROOT}/networking/firewalls/${encodeURIComponent(
      firewallID
    )}/devices/${encodeURIComponent(deviceID)}`
  )
);

// src/images/images.ts
import {
  createImageSchema,
  updateImageSchema,
  uploadImageSchema
} from "@linode/validation/lib/images.schema";
var getImage = (imageId) => request_default(
  setURL(`${API_ROOT}/images/${encodeURIComponent(imageId)}`),
  setMethod("GET")
);
var getImages = (params = {}, filters = {}) => request_default(
  setURL(`${API_ROOT}/images`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filters)
);
var createImage = (diskId, label, description, cloud_init) => {
  const data = __spreadValues(__spreadValues(__spreadValues({
    disk_id: diskId
  }, label && { label }), description && { description }), cloud_init && { cloud_init });
  return request_default(
    setURL(`${API_ROOT}/images`),
    setMethod("POST"),
    setData(data, createImageSchema)
  );
};
var updateImage = (imageId, label, description) => {
  const data = __spreadValues(__spreadValues({}, label && { label }), description && { description });
  return request_default(
    setURL(`${API_ROOT}/images/${encodeURIComponent(imageId)}`),
    setMethod("PUT"),
    setData(data, updateImageSchema)
  );
};
var deleteImage = (imageId) => {
  return request_default(
    setURL(`${API_ROOT}/images/${encodeURIComponent(imageId)}`),
    setMethod("DELETE")
  );
};
var uploadImage = (data) => {
  return request_default(
    setURL(`${API_ROOT}/images/upload`),
    setMethod("POST"),
    setData(data, uploadImageSchema)
  );
};

// src/kubernetes/kubernetes.ts
import { createKubeClusterSchema } from "@linode/validation/lib/kubernetes.schema";
var getKubernetesClusters = (params, filters) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filters),
  setURL(`${API_ROOT}/lke/clusters`)
);
var getKubernetesCluster = (clusterID) => request_default(
  setMethod("GET"),
  setURL(`${API_ROOT}/lke/clusters/${encodeURIComponent(clusterID)}`)
);
var createKubernetesCluster = (data) => request_default(
  setMethod("POST"),
  setURL(`${API_ROOT}/lke/clusters`),
  setData(data, createKubeClusterSchema)
);
var updateKubernetesCluster = (clusterID, data) => request_default(
  setMethod("PUT"),
  setURL(`${API_ROOT}/lke/clusters/${encodeURIComponent(clusterID)}`),
  setData(data)
);
var deleteKubernetesCluster = (clusterID) => request_default(
  setMethod("DELETE"),
  setURL(`${API_ROOT}/lke/clusters/${encodeURIComponent(clusterID)}`)
);
var getKubeConfig = (clusterId) => request_default(
  setMethod("GET"),
  setURL(
    `${API_ROOT}/lke/clusters/${encodeURIComponent(clusterId)}/kubeconfig`
  )
);
var resetKubeConfig = (clusterId) => request_default(
  setMethod("DELETE"),
  setURL(
    `${API_ROOT}/lke/clusters/${encodeURIComponent(clusterId)}/kubeconfig`
  )
);
var getKubernetesVersions = (params, filters) => request_default(
  setMethod("GET"),
  setXFilter(filters),
  setParams(params),
  setURL(`${API_ROOT}/lke/versions`)
);
var getKubernetesVersion = (versionID) => request_default(
  setMethod("GET"),
  setURL(`${API_ROOT}/lke/versions/${encodeURIComponent(versionID)}`)
);
var getKubernetesClusterEndpoints = (clusterID, params, filters) => request_default(
  setMethod("GET"),
  setXFilter(filters),
  setParams(params),
  setURL(
    `${API_ROOT}/lke/clusters/${encodeURIComponent(clusterID)}/api-endpoints`
  )
);
var getKubernetesClusterDashboard = (clusterID) => request_default(
  setMethod("GET"),
  setURL(
    `${API_ROOT}/lke/clusters/${encodeURIComponent(clusterID)}/dashboard`
  )
);
var recycleClusterNodes = (clusterID) => request_default(
  setMethod("POST"),
  setURL(`${API_ROOT}/lke/clusters/${encodeURIComponent(clusterID)}/recycle`)
);

// src/kubernetes/nodePools.ts
import { nodePoolSchema } from "@linode/validation/lib/kubernetes.schema";
var getNodePools = (clusterID, params, filters) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filters),
  setURL(`${API_ROOT}/lke/clusters/${encodeURIComponent(clusterID)}/pools`)
);
var getNodePool = (clusterID, nodePoolID) => request_default(
  setMethod("GET"),
  setURL(
    `${API_ROOT}/lke/clusters/${encodeURIComponent(
      clusterID
    )}/pools/${encodeURIComponent(nodePoolID)}`
  )
);
var createNodePool = (clusterID, data) => request_default(
  setMethod("POST"),
  setURL(`${API_ROOT}/lke/clusters/${encodeURIComponent(clusterID)}/pools`),
  setData(data, nodePoolSchema)
);
var updateNodePool = (clusterID, nodePoolID, data) => request_default(
  setMethod("PUT"),
  setURL(
    `${API_ROOT}/lke/clusters/${encodeURIComponent(
      clusterID
    )}/pools/${encodeURIComponent(nodePoolID)}`
  ),
  setData(data, nodePoolSchema)
);
var deleteNodePool = (clusterID, nodePoolID) => request_default(
  setMethod("DELETE"),
  setURL(
    `${API_ROOT}/lke/clusters/${encodeURIComponent(
      clusterID
    )}/pools/${encodeURIComponent(nodePoolID)}`
  )
);
var recycleAllNodes = (clusterID, nodePoolID) => request_default(
  setMethod("POST"),
  setURL(
    `${API_ROOT}/lke/clusters/${encodeURIComponent(
      clusterID
    )}/pools/${encodeURIComponent(nodePoolID)}/recycle`
  )
);
var recycleNode = (clusterID, nodeID) => request_default(
  setMethod("POST"),
  setURL(
    `${API_ROOT}/lke/clusters/${encodeURIComponent(
      clusterID
    )}/nodes/${encodeURIComponent(nodeID)}/recycle`
  )
);

// src/linodes/actions.ts
import { RebuildLinodeSchema } from "@linode/validation/lib/linodes.schema";
var linodeBoot = (linodeId, config_id) => request_default(
  setURL(`${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/boot`),
  setMethod("POST"),
  setData({ config_id })
);
var linodeReboot = (linodeId, config_id) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/reboot`
  ),
  setMethod("POST"),
  setData({ config_id })
);
var linodeShutdown = (linodeId) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/shutdown`
  ),
  setMethod("POST")
);
var resizeLinode = (linodeId, data) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/resize`
  ),
  setMethod("POST"),
  setData(data)
);
var rebuildLinode = (linodeId, data) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/rebuild`
  ),
  setMethod("POST"),
  setData(data, RebuildLinodeSchema)
);
var rescueLinode = (linodeId, devices) => {
  const _devices = __spreadValues({}, devices);
  delete _devices["sdh"];
  return request_default(
    setURL(
      `${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/rescue`
    ),
    setMethod("POST"),
    setData({ devices: _devices })
  );
};
var rescueMetalLinode = (linodeId) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/rescue`
  ),
  setMethod("POST")
);
var cloneLinode = (sourceLinodeId, data) => {
  return request_default(
    setURL(
      `${API_ROOT}/linode/instances/${encodeURIComponent(sourceLinodeId)}/clone`
    ),
    setMethod("POST"),
    setData(data)
  );
};
var startMutation = (linodeID) => {
  return request_default(
    setURL(
      `${API_ROOT}/linode/instances/${encodeURIComponent(linodeID)}/mutate`
    ),
    setMethod("POST")
  );
};
var scheduleOrQueueMigration = (linodeID, payload) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(linodeID)}/migrate`
  ),
  setData(payload || {}),
  setMethod("POST")
);

// src/linodes/backups.ts
import { CreateSnapshotSchema } from "@linode/validation/lib/linodes.schema";
var getLinodeBackups = (id) => request_default(
  setURL(`${API_ROOT}/linode/instances/${encodeURIComponent(id)}/backups`),
  setMethod("GET")
);
var enableBackups = (linodeId) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/backups/enable`
  ),
  setMethod("POST")
);
var cancelBackups = (linodeId) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/backups/cancel`
  ),
  setMethod("POST")
);
var takeSnapshot = (linodeId, label) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/backups`
  ),
  setMethod("POST"),
  setData({ label }, CreateSnapshotSchema)
);
var restoreBackup = (linodeId, backupId, targetLinodeId, overwrite) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/backups/${encodeURIComponent(backupId)}/restore`
  ),
  setMethod("POST"),
  setData({ linode_id: targetLinodeId, overwrite })
);

// src/linodes/configs.ts
import {
  CreateLinodeConfigSchema,
  UpdateConfigInterfaceOrderSchema,
  UpdateConfigInterfaceSchema,
  UpdateLinodeConfigSchema,
  LinodeInterfaceSchema
} from "@linode/validation/lib/linodes.schema";
var getLinodeConfigs = (linodeId, params, filters) => request_default(
  setURL(`${API_ROOT}/linode/instances/${linodeId}/configs`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filters)
);
var getLinodeConfig = (linodeId, configId) => request_default(
  setURL(`${API_ROOT}/linode/instances/${linodeId}/configs/${configId}`),
  setMethod("GET")
);
var createLinodeConfig = (linodeId, data) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/configs`
  ),
  setMethod("POST"),
  setData(data, CreateLinodeConfigSchema)
);
var deleteLinodeConfig = (linodeId, configId) => request_default(
  setMethod("DELETE"),
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/configs/${encodeURIComponent(configId)}`
  )
);
var updateLinodeConfig = (linodeId, configId, data) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/configs/${encodeURIComponent(configId)}`
  ),
  setMethod("PUT"),
  setData(data, UpdateLinodeConfigSchema)
);
var getConfigInterfaces = (linodeId, configId) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/configs/${encodeURIComponent(configId)}/interfaces`
  ),
  setMethod("GET")
);
var getConfigInterface = (linodeId, configId, interfaceId) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/configs/${encodeURIComponent(
      configId
    )}/interfaces/${encodeURIComponent(interfaceId)}`
  ),
  setMethod("GET")
);
var appendConfigInterface = (linodeId, configId, data) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/configs/${encodeURIComponent(configId)}/interfaces`
  ),
  setMethod("POST"),
  setData(data, LinodeInterfaceSchema)
);
var updateConfigInterface = (linodeId, configId, interfaceId, data) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/configs/${encodeURIComponent(
      configId
    )}/interfaces/${encodeURIComponent(interfaceId)}`
  ),
  setMethod("PUT"),
  setData(data, UpdateConfigInterfaceSchema)
);
var updateLinodeConfigOrder = (linodeId, configId, data) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/configs/${encodeURIComponent(configId)}/interfaces/order`
  ),
  setMethod("POST"),
  setData(data, UpdateConfigInterfaceOrderSchema)
);
var deleteLinodeConfigInterface = (linodeId, configId, interfaceId) => request_default(
  setMethod("DELETE"),
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/configs/${encodeURIComponent(
      configId
    )}/interfaces/${encodeURIComponent(interfaceId)}`
  )
);

// src/linodes/disks.ts
import {
  CreateLinodeDiskSchema,
  ResizeLinodeDiskSchema,
  UpdateLinodePasswordSchema
} from "@linode/validation/lib/linodes.schema";
var getLinodeDisks = (linodeId, params, filter) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/disks`
  ),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var createLinodeDisk = (linodeId, data) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/disks`
  ),
  setMethod("POST"),
  setData(data, CreateLinodeDiskSchema)
);
var getLinodeDisk = (linodeId, diskId) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/disks/${encodeURIComponent(diskId)}`
  ),
  setMethod("GET")
);
var updateLinodeDisk = (linodeId, diskId, data) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/disks/${encodeURIComponent(diskId)}`
  ),
  setMethod("PUT"),
  setData(data)
);
var resizeLinodeDisk = (linodeId, diskId, size) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/disks/${encodeURIComponent(diskId)}/resize`
  ),
  setMethod("POST"),
  setData({ size }, ResizeLinodeDiskSchema)
);
var cloneLinodeDisk = (linodeId, diskId) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/disks/${encodeURIComponent(diskId)}/clone`
  ),
  setMethod("POST")
);
var deleteLinodeDisk = (linodeId, diskId) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/disks/${encodeURIComponent(diskId)}`
  ),
  setMethod("DELETE")
);
var changeLinodeDiskPassword = (linodeId, diskId, password) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/disks/${encodeURIComponent(diskId)}/password`
  ),
  setMethod("POST"),
  setData({ password }, UpdateLinodePasswordSchema)
);

// src/linodes/info.ts
var getLinodeStats = (linodeId) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/stats`
  ),
  setMethod("GET")
);
var getLinodeStatsByDate = (linodeId, year, month) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/stats/${encodeURIComponent(year)}/${encodeURIComponent(month)}`
  ),
  setMethod("GET")
);
var getLinodeTransfer = (linodeId) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/transfer`
  ),
  setMethod("GET")
);
var getLinodeTransferByDate = (linodeId, year, month) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(
      linodeId
    )}/transfer/${encodeURIComponent(year)}/${encodeURIComponent(month)}`
  ),
  setMethod("GET")
);
var getLinodeKernels = (params, filter) => request_default(
  setURL(`${API_ROOT}/linode/kernels`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getLinodeKernel = (kernelId) => request_default(
  setURL(`${API_ROOT}/linode/kernels/${encodeURIComponent(kernelId)}`),
  setMethod("GET")
);
var getLinodeTypes = (params) => request_default(
  setURL(`${API_ROOT}/linode/types`),
  setMethod("GET"),
  setParams(params)
);
var getType = (typeId) => request_default(
  setURL(`${API_ROOT}/linode/types/${encodeURIComponent(typeId)}`),
  setMethod("GET")
);
var getDeprecatedLinodeTypes = () => request_default(
  setURL(`${API_ROOT}/linode/types-legacy`),
  setMethod("GET")
);

// src/linodes/ips.ts
import { IPAllocationSchema } from "@linode/validation/lib/linodes.schema";
var getLinodeIPs = (id) => request_default(
  setURL(`${API_ROOT}/linode/instances/${encodeURIComponent(id)}/ips`),
  setMethod("GET")
);
var allocateIPAddress = (linodeID, data) => request_default(
  setURL(`${API_ROOT}/linode/instances/${encodeURIComponent(linodeID)}/ips`),
  setMethod("POST"),
  setData(data, IPAllocationSchema)
);
var removeIPAddress = (payload) => {
  return request_default(
    setURL(
      `${API_ROOT}/linode/instances/${encodeURIComponent(
        payload.linodeID
      )}/ips/${encodeURIComponent(payload.address)}`
    ),
    setMethod("DELETE")
  );
};
var removeIPv6Range = (payload) => {
  return request_default(
    setURL(
      `${API_ROOT}/networking/ipv6/ranges/${encodeURIComponent(payload.range)}`
    ),
    setMethod("DELETE")
  );
};

// src/linodes/linodes.ts
import {
  CreateLinodeSchema,
  UpdateLinodeSchema
} from "@linode/validation/lib/linodes.schema";
var getLinode = (linodeId) => request_default(
  setURL(`${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}`),
  setMethod("GET")
);
var getLinodeLishToken = (linodeId) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/lish_token`
  ),
  setMethod("POST")
);
var getLinodeVolumes = (linodeId, params = {}, filter = {}) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/volumes`
  ),
  setMethod("GET"),
  setXFilter(filter),
  setParams(params)
);
var getLinodes = (params, filter) => request_default(
  setURL(`${API_ROOT}/linode/instances/`),
  setMethod("GET"),
  setXFilter(filter),
  setParams(params)
);
var createLinode = (data) => request_default(
  setURL(`${API_ROOT}/linode/instances`),
  setMethod("POST"),
  setData(data, CreateLinodeSchema)
);
var updateLinode = (linodeId, values) => request_default(
  setURL(`${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}`),
  setMethod("PUT"),
  setData(values, UpdateLinodeSchema)
);
var deleteLinode = (linodeId) => request_default(
  setURL(`${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}`),
  setMethod("DELETE")
);
var changeLinodePassword = (linodeId, root_pass) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/password`
  ),
  setData({ root_pass }),
  setMethod("POST")
);
var getLinodeFirewalls = (linodeId, params, filter) => request_default(
  setURL(
    `${API_ROOT}/linode/instances/${encodeURIComponent(linodeId)}/firewalls`
  ),
  setMethod("GET"),
  setXFilter(filter),
  setParams(params)
);

// src/longview/longview.ts
import { longviewClientCreate } from "@linode/validation/lib/longview.schema";
var createLongviewClient = (label) => {
  return request_default(
    setURL(`${API_ROOT}/longview/clients`),
    setData(
      {
        label
      },
      longviewClientCreate
    ),
    setMethod("POST")
  );
};
var getLongviewClients = (params, filter) => request_default(
  setURL(`${API_ROOT}/longview/clients`),
  setParams(params),
  setXFilter(filter),
  setMethod("GET")
);
var deleteLongviewClient = (id) => request_default(
  setURL(`${API_ROOT}/longview/clients/${encodeURIComponent(id)}`),
  setMethod("DELETE")
);
var updateLongviewClient = (id, label) => {
  return request_default(
    setURL(`${API_ROOT}/longview/clients/${encodeURIComponent(id)}`),
    setData(
      {
        label
      },
      longviewClientCreate
    ),
    setMethod("PUT")
  );
};
var getLongviewSubscriptions = () => request_default(
  setURL(`${API_ROOT}/longview/subscriptions`),
  setMethod("GET")
);
var getActiveLongviewPlan = () => request_default(
  setURL(`${API_ROOT}/longview/plan`),
  setMethod("GET")
);
var updateActiveLongviewPlan = (plan) => request_default(
  setURL(`${API_ROOT}/longview/plan`),
  setData(plan),
  setMethod("PUT")
);

// src/managed/managed.ts
import {
  createContactSchema,
  createCredentialSchema,
  createServiceMonitorSchema,
  updateCredentialSchema,
  updateManagedLinodeSchema,
  updatePasswordSchema
} from "@linode/validation/lib/managed.schema";
var enableManaged = () => request_default(
  setMethod("POST"),
  setURL(`${API_ROOT}/account/settings/managed-enable`)
);
var getServices = (params, filters) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filters),
  setURL(`${API_ROOT}/managed/services`)
);
var disableServiceMonitor = (serviceID) => request_default(
  setMethod("POST"),
  setURL(
    `${API_ROOT}/managed/services/${encodeURIComponent(serviceID)}/disable`
  )
);
var enableServiceMonitor = (serviceID) => request_default(
  setMethod("POST"),
  setURL(
    `${API_ROOT}/managed/services/${encodeURIComponent(serviceID)}/enable`
  )
);
var deleteServiceMonitor = (serviceID) => request_default(
  setMethod("DELETE"),
  setURL(`${API_ROOT}/managed/services/${encodeURIComponent(serviceID)}`)
);
var getLinodeSettings = (params, filters) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filters),
  setURL(`${API_ROOT}/managed/linode-settings`)
);
var createServiceMonitor = (data) => request_default(
  setMethod("POST"),
  setURL(`${API_ROOT}/managed/services`),
  setData(data, createServiceMonitorSchema)
);
var updateServiceMonitor = (monitorID, data) => request_default(
  setMethod("PUT"),
  setURL(`${API_ROOT}/managed/services/${encodeURIComponent(monitorID)}`),
  setData(data, createServiceMonitorSchema)
);
var getCredentials = (params, filters) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filters),
  setURL(`${API_ROOT}/managed/credentials`)
);
var updateCredential = (credentialID, data) => request_default(
  setMethod("PUT"),
  setData(data, updateCredentialSchema),
  setURL(
    `${API_ROOT}/managed/credentials/${encodeURIComponent(credentialID)}`
  )
);
var updatePassword = (credentialID, data) => request_default(
  setMethod("POST"),
  setData(data, updatePasswordSchema),
  setURL(
    `${API_ROOT}/managed/credentials/${encodeURIComponent(
      credentialID
    )}/update`
  )
);
var deleteCredential = (credentialID) => request_default(
  setMethod("POST"),
  setURL(
    `${API_ROOT}/managed/credentials/${encodeURIComponent(
      credentialID
    )}/revoke`
  )
);
var createCredential = (data) => request_default(
  setMethod("POST"),
  setURL(`${API_ROOT}/managed/credentials`),
  setData(data, createCredentialSchema)
);
var getSSHPubKey = () => request_default(
  setMethod("GET"),
  setURL(`${API_ROOT}/managed/credentials/sshkey`)
);
var updateLinodeSettings = (linodeId, data) => request_default(
  setURL(
    `${API_ROOT}/managed/linode-settings/${encodeURIComponent(linodeId)}`
  ),
  setMethod("PUT"),
  setData(data, updateManagedLinodeSchema)
);
var getManagedContacts = (params, filters) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filters),
  setURL(`${API_ROOT}/managed/contacts`)
);
var createContact = (data) => request_default(
  setMethod("POST"),
  setURL(`${API_ROOT}/managed/contacts`),
  setData(data, createContactSchema)
);
var updateContact = (contactId, data) => request_default(
  setMethod("PUT"),
  setURL(`${API_ROOT}/managed/contacts/${encodeURIComponent(contactId)}`),
  setData(data, createContactSchema)
);
var deleteContact = (contactId) => request_default(
  setMethod("DELETE"),
  setURL(`${API_ROOT}/managed/contacts/${encodeURIComponent(contactId)}`)
);
var getManagedIssues = () => request_default(
  setMethod("GET"),
  setURL(`${API_ROOT}/managed/issues`)
);
var getManagedStats = () => request_default(setMethod("GET"), setURL(`${API_ROOT}/managed/stats`));

// src/networking/networking.ts
import {
  allocateIPSchema,
  assignAddressesSchema,
  shareAddressesSchema,
  updateIPSchema
} from "@linode/validation/lib/networking.schema";
var getIPs = (params, filters) => request_default(
  setURL(`${API_ROOT}/networking/ips`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filters)
);
var getIP = (address) => request_default(
  setURL(`${API_ROOT}/networking/ips/${encodeURIComponent(address)}`),
  setMethod("GET")
);
var updateIP = (address, rdns = null) => request_default(
  setURL(`${API_ROOT}/networking/ips/${encodeURIComponent(address)}`),
  setData({ rdns }, updateIPSchema),
  setMethod("PUT")
);
var allocateIp = (payload) => request_default(
  setURL(`${API_ROOT}/networking/ips/`),
  setData(payload, allocateIPSchema),
  setMethod("POST")
);
var assignAddresses = (payload) => request_default(
  setURL(`${API_ROOT}/networking/ips/assign`),
  setMethod("POST"),
  setData(payload, assignAddressesSchema)
);
var shareAddressesv4 = (payload) => request_default(
  setURL(`${API_ROOT}/networking/ipv4/share`),
  setMethod("POST"),
  setData(payload, shareAddressesSchema)
);
var shareAddresses = (payload) => request_default(
  setURL(`${BETA_API_ROOT}/networking/ips/share`),
  setMethod("POST"),
  setData(payload, shareAddressesSchema)
);
var getIPv6Pools = (params) => request_default(
  setURL(`${API_ROOT}/networking/ipv6/pools`),
  setMethod("GET"),
  setParams(params)
);
var getIPv6Ranges = (params, filter) => request_default(
  setURL(`${API_ROOT}/networking/ipv6/ranges`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getIPv6RangeInfo = (range, params) => request_default(
  setURL(`${API_ROOT}/networking/ipv6/ranges/${encodeURIComponent(range)}`),
  setMethod("GET"),
  setParams(params)
);
var createIPv6Range = (payload) => {
  return request_default(
    setURL(`${API_ROOT}/networking/ipv6/ranges`),
    setMethod("POST"),
    setData(payload)
  );
};

// src/nodebalancers/nodebalancers.ts
import {
  NodeBalancerSchema,
  UpdateNodeBalancerSchema
} from "@linode/validation/lib/nodebalancers.schema";

// src/nodebalancers/utils.ts
var combineConfigNodeAddressAndPort = (data) => __spreadProps(__spreadValues({}, data), {
  nodes: data.nodes.map((n) => ({
    address: `${n.address}:${n.port}`,
    label: n.label,
    mode: n.mode,
    weight: n.weight
  }))
});
var combineNodeBalancerConfigNodeAddressAndPort = (data) => __spreadProps(__spreadValues({}, data), {
  configs: data.configs.map((c) => __spreadProps(__spreadValues({}, c), {
    nodes: c.nodes.map((n) => ({
      address: `${n.address}:${n.port}`,
      label: n.label,
      mode: n.mode,
      weight: n.weight
    }))
  }))
});
var mergeAddressAndPort = (node) => __spreadProps(__spreadValues({}, node), {
  address: `${node.address}:${node.port}`
});

// src/nodebalancers/nodebalancers.ts
var getNodeBalancers = (params, filters) => request_default(
  setURL(`${API_ROOT}/nodebalancers`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filters)
);
var getNodeBalancer = (nodeBalancerId) => request_default(
  setURL(`${API_ROOT}/nodebalancers/${encodeURIComponent(nodeBalancerId)}`),
  setMethod("GET")
);
var updateNodeBalancer = (nodeBalancerId, data) => request_default(
  setURL(`${API_ROOT}/nodebalancers/${encodeURIComponent(nodeBalancerId)}`),
  setMethod("PUT"),
  setData(data, UpdateNodeBalancerSchema)
);
var createNodeBalancer = (data) => request_default(
  setMethod("POST"),
  setURL(`${API_ROOT}/nodebalancers`),
  setData(
    data,
    NodeBalancerSchema,
    combineNodeBalancerConfigNodeAddressAndPort
  )
);
var deleteNodeBalancer = (nodeBalancerId) => request_default(
  setMethod("DELETE"),
  setURL(`${API_ROOT}/nodebalancers/${encodeURIComponent(nodeBalancerId)}`)
);
var getNodeBalancerStats = (nodeBalancerId) => {
  return request_default(
    setURL(
      `${API_ROOT}/nodebalancers/${encodeURIComponent(nodeBalancerId)}/stats`
    ),
    setMethod("GET")
  );
};
var getNodeBalancerFirewalls = (nodeBalancerId, params, filter) => request_default(
  setURL(
    `${API_ROOT}/nodebalancers/${encodeURIComponent(
      nodeBalancerId
    )}/firewalls`
  ),
  setMethod("GET"),
  setXFilter(filter),
  setParams(params)
);
var getNodeBalancerTypes = (params) => request_default(
  setURL(`${API_ROOT}/nodebalancers/types`),
  setMethod("GET"),
  setParams(params)
);

// src/nodebalancers/nodebalancer-configs.ts
import {
  createNodeBalancerConfigSchema,
  UpdateNodeBalancerConfigSchema
} from "@linode/validation/lib/nodebalancers.schema";
var getNodeBalancerConfigs = (nodeBalancerId, params) => request_default(
  setURL(
    `${API_ROOT}/nodebalancers/${encodeURIComponent(nodeBalancerId)}/configs`
  ),
  setMethod("GET"),
  setParams(params)
);
var getNodeBalancerConfig = (nodeBalancerId, configId) => request_default(
  setURL(
    `${API_ROOT}/nodebalancers/${encodeURIComponent(
      nodeBalancerId
    )}/configs/${encodeURIComponent(configId)}`
  ),
  setMethod("GET")
);
var createNodeBalancerConfig = (nodeBalancerId, data) => request_default(
  setMethod("POST"),
  setURL(
    `${API_ROOT}/nodebalancers/${encodeURIComponent(nodeBalancerId)}/configs`
  ),
  setData(
    data,
    createNodeBalancerConfigSchema,
    combineConfigNodeAddressAndPort
  )
);
var updateNodeBalancerConfig = (nodeBalancerId, configId, data) => request_default(
  setMethod("PUT"),
  setURL(
    `${API_ROOT}/nodebalancers/${encodeURIComponent(
      nodeBalancerId
    )}/configs/${encodeURIComponent(configId)}`
  ),
  setData(data, UpdateNodeBalancerConfigSchema)
);
var deleteNodeBalancerConfig = (nodeBalancerId, configId) => request_default(
  setMethod("DELETE"),
  setURL(
    `${API_ROOT}/nodebalancers/${encodeURIComponent(
      nodeBalancerId
    )}/configs/${encodeURIComponent(configId)}`
  )
);

// src/nodebalancers/nodebalancer-config-nodes.ts
import { nodeBalancerConfigNodeSchema } from "@linode/validation/lib/nodebalancers.schema";
var getNodeBalancerConfigNodes = (nodeBalancerId, configId) => request_default(
  setMethod("GET"),
  setURL(
    `${API_ROOT}/nodebalancers/${encodeURIComponent(
      nodeBalancerId
    )}/configs/${encodeURIComponent(configId)}/nodes`
  )
);
var getNodeBalancerConfigNode = (nodeBalancerId, configId, nodeId) => request_default(
  setMethod("GET"),
  setURL(
    `${API_ROOT}/nodebalancers/${encodeURIComponent(
      nodeBalancerId
    )}/configs/${encodeURIComponent(configId)}/nodes/${encodeURIComponent(
      nodeId
    )}`
  )
);
var createNodeBalancerConfigNode = (nodeBalancerId, configId, data) => request_default(
  setMethod("POST"),
  setURL(
    `${API_ROOT}/nodebalancers/${encodeURIComponent(
      nodeBalancerId
    )}/configs/${encodeURIComponent(configId)}/nodes`
  ),
  setData(data, nodeBalancerConfigNodeSchema, mergeAddressAndPort)
);
var updateNodeBalancerConfigNode = (nodeBalancerId, configId, nodeId, data) => request_default(
  setMethod("PUT"),
  setURL(
    `${API_ROOT}/nodebalancers/${encodeURIComponent(
      nodeBalancerId
    )}/configs/${encodeURIComponent(configId)}/nodes/${encodeURIComponent(
      nodeId
    )}`
  ),
  setData(data, nodeBalancerConfigNodeSchema, mergeAddressAndPort)
);
var deleteNodeBalancerConfigNode = (nodeBalancerId, configId, nodeId) => request_default(
  setMethod("DELETE"),
  setURL(
    `${API_ROOT}/nodebalancers/${encodeURIComponent(
      nodeBalancerId
    )}/configs/${encodeURIComponent(configId)}/nodes/${encodeURIComponent(
      nodeId
    )}`
  )
);

// src/object-storage/account.ts
var cancelObjectStorage = () => request_default(setMethod("POST"), setURL(`${API_ROOT}/object-storage/cancel`));

// src/object-storage/buckets.ts
import {
  CreateBucketSchema,
  UpdateBucketAccessSchema,
  UploadCertificateSchema
} from "@linode/validation/lib/buckets.schema";
var getBucket = (clusterId, bucketName) => request_default(
  setMethod("GET"),
  setURL(
    `${API_ROOT}/object-storage/buckets/${encodeURIComponent(
      clusterId
    )}/${encodeURIComponent(bucketName)}`
  )
);
var getBuckets = (params, filters) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filters),
  setURL(`${API_ROOT}/object-storage/buckets`)
);
var getBucketsInCluster = (clusterId, params, filters) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filters),
  setURL(
    `${API_ROOT}/object-storage/buckets/${encodeURIComponent(clusterId)}`
  )
);
var getBucketsInRegion = (regionId, params, filters) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filters),
  setURL(`${API_ROOT}/object-storage/buckets/${encodeURIComponent(regionId)}`)
);
var createBucket = (data) => request_default(
  setURL(`${API_ROOT}/object-storage/buckets`),
  setMethod("POST"),
  setData(data, CreateBucketSchema)
);
var deleteBucket = ({
  cluster,
  label
}) => request_default(
  setURL(
    `${API_ROOT}/object-storage/buckets/${encodeURIComponent(
      cluster
    )}/${encodeURIComponent(label)}`
  ),
  setMethod("DELETE")
);
var deleteBucketWithRegion = ({
  region,
  label
}) => request_default(
  setURL(
    `${API_ROOT}/object-storage/buckets/${encodeURIComponent(
      region
    )}/${encodeURIComponent(label)}`
  ),
  setMethod("DELETE")
);
var getObjectList = (clusterId, bucketName, params) => request_default(
  setMethod("GET"),
  setParams(params),
  setURL(
    `${API_ROOT}/object-storage/buckets/${encodeURIComponent(
      clusterId
    )}/${encodeURIComponent(bucketName)}/object-list`
  )
);
var uploadSSLCert = (clusterId, bucketName, data) => request_default(
  setMethod("POST"),
  setData(data, UploadCertificateSchema),
  setURL(
    `${API_ROOT}/object-storage/buckets/${encodeURIComponent(
      clusterId
    )}/${encodeURIComponent(bucketName)}/ssl`
  )
);
var getSSLCert = (clusterId, bucketName) => request_default(
  setMethod("GET"),
  setURL(
    `${API_ROOT}/object-storage/buckets/${encodeURIComponent(
      clusterId
    )}/${encodeURIComponent(bucketName)}/ssl`
  )
);
var deleteSSLCert = (clusterId, bucketName) => request_default(
  setMethod("DELETE"),
  setURL(
    `${API_ROOT}/object-storage/buckets/${encodeURIComponent(
      clusterId
    )}/${encodeURIComponent(bucketName)}/ssl`
  )
);
var getBucketAccess = (clusterId, bucketName) => request_default(
  setMethod("GET"),
  setURL(
    `${API_ROOT}/object-storage/buckets/${encodeURIComponent(
      clusterId
    )}/${encodeURIComponent(bucketName)}/access`
  )
);
var updateBucketAccess = (clusterId, bucketName, data) => request_default(
  setMethod("PUT"),
  setURL(
    `${API_ROOT}/object-storage/buckets/${encodeURIComponent(
      clusterId
    )}/${encodeURIComponent(bucketName)}/access`
  ),
  setData(data, UpdateBucketAccessSchema)
);

// src/object-storage/clusters.ts
var getClusters = (params, filters) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filters),
  setURL(`${API_ROOT}/object-storage/clusters`)
);

// src/object-storage/objects.ts
var getObjectURL = (clusterId, bucketName, name, method, options) => request_default(
  setMethod("POST"),
  setURL(
    `${API_ROOT}/object-storage/buckets/${encodeURIComponent(
      clusterId
    )}/${encodeURIComponent(bucketName)}/object-url`
  ),
  setData(__spreadValues({ name, method }, options))
);
var getObjectACL = (clusterId, bucketName, name) => request_default(
  setMethod("GET"),
  setURL(
    `${API_ROOT}/object-storage/buckets/${encodeURIComponent(
      clusterId
    )}/${encodeURIComponent(bucketName)}/object-acl?name=${encodeURIComponent(
      name
    )}`
  )
);
var updateObjectACL = (clusterId, bucketName, name, acl) => request_default(
  setMethod("PUT"),
  setURL(
    `${API_ROOT}/object-storage/buckets/${encodeURIComponent(
      clusterId
    )}/${encodeURIComponent(bucketName)}/object-acl`
  ),
  setData({ acl, name })
);

// src/object-storage/objectStorageKeys.ts
import {
  createObjectStorageKeysSchema,
  updateObjectStorageKeysSchema
} from "@linode/validation/lib/objectStorageKeys.schema";
var getObjectStorageKeys = (params, filters) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filters),
  setURL(`${API_ROOT}/object-storage/keys`)
);
var createObjectStorageKeys = (data) => request_default(
  setMethod("POST"),
  setURL(`${API_ROOT}/object-storage/keys`),
  setData(data, createObjectStorageKeysSchema)
);
var updateObjectStorageKey = (id, data) => request_default(
  setMethod("PUT"),
  setURL(`${API_ROOT}/object-storage/keys/${encodeURIComponent(id)}`),
  setData(data, updateObjectStorageKeysSchema)
);
var revokeObjectStorageKey = (id) => request_default(
  setMethod("DELETE"),
  setURL(`${API_ROOT}/object-storage/keys/${encodeURIComponent(id)}`)
);

// src/placement-groups/types.ts
var AFFINITY_TYPES = {
  "affinity:local": "Affinity",
  "anti_affinity:local": "Anti-affinity"
};

// src/placement-groups/placement-groups.ts
import {
  createPlacementGroupSchema,
  updatePlacementGroupSchema
} from "@linode/validation";
var getPlacementGroups = (params, filter) => request_default(
  setURL(`${API_ROOT}/placement/groups`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getPlacementGroup = (placementGroupId) => request_default(
  setURL(
    `${API_ROOT}/placement/groups/${encodeURIComponent(placementGroupId)}`
  ),
  setMethod("GET")
);
var createPlacementGroup = (data) => request_default(
  setURL(`${API_ROOT}/placement/groups`),
  setMethod("POST"),
  setData(data, createPlacementGroupSchema)
);
var updatePlacementGroup = (placementGroupId, data) => request_default(
  setURL(
    `${API_ROOT}/placement/groups/${encodeURIComponent(placementGroupId)}`
  ),
  setMethod("PUT"),
  setData(data, updatePlacementGroupSchema)
);
var deletePlacementGroup = (placementGroupId) => request_default(
  setURL(
    `${API_ROOT}/placement/groups/${encodeURIComponent(placementGroupId)}`
  ),
  setMethod("DELETE")
);
var assignLinodesToPlacementGroup = (placementGroupId, payload) => request_default(
  setURL(
    `${API_ROOT}/placement/groups/${encodeURIComponent(
      placementGroupId
    )}/assign`
  ),
  setMethod("POST"),
  setData(payload)
);
var unassignLinodesFromPlacementGroup = (placementGroupId, payload) => request_default(
  setURL(
    `${API_ROOT}/placement/groups/${encodeURIComponent(
      placementGroupId
    )}/unassign`
  ),
  setMethod("POST"),
  setData(payload)
);

// src/profile/twofactor.ts
import { enableTwoFactorSchema } from "@linode/validation/lib/twofactor.schema";
var getTFAToken = () => request_default(setMethod("POST"), setURL(`${API_ROOT}/profile/tfa-enable`));
var disableTwoFactor = () => request_default(setMethod("POST"), setURL(`${API_ROOT}/profile/tfa-disable`));
var confirmTwoFactor = (tfa_code) => request_default(
  setMethod("POST"),
  setURL(`${API_ROOT}/profile/tfa-enable-confirm`),
  setData({ tfa_code }, enableTwoFactorSchema)
);

// src/profile/sshkeys.ts
import {
  createSSHKeySchema,
  updateSSHKeySchema
} from "@linode/validation/lib/profile.schema";
var getSSHKeys = (params, filters) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filters),
  setURL(`${API_ROOT}/profile/sshkeys`)
);
var getSSHKey = (keyId) => request_default(
  setMethod("GET"),
  setURL(`${API_ROOT}/profile/sshkeys/${encodeURIComponent(keyId)}`)
);
var createSSHKey = (data) => request_default(
  setMethod("POST"),
  setURL(`${API_ROOT}/profile/sshkeys`),
  setData(data, createSSHKeySchema)
);
var updateSSHKey = (keyId, data) => request_default(
  setMethod("PUT"),
  setURL(`${API_ROOT}/profile/sshkeys/${encodeURIComponent(keyId)}`),
  setData(data, updateSSHKeySchema)
);
var deleteSSHKey = (keyId) => request_default(
  setMethod("DELETE"),
  setURL(`${API_ROOT}/profile/sshkeys/${encodeURIComponent(keyId)}`)
);

// src/profile/profile.ts
import {
  SendCodeToPhoneNumberSchema,
  VerifyPhoneNumberCodeSchema,
  SecurityQuestionsSchema
} from "@linode/validation/lib/profile.schema";
import { updateProfileSchema } from "@linode/validation/lib/profile.schema";
var getProfile = ({ headers } = {}) => {
  return request_default(
    setURL(`${API_ROOT}/profile`),
    setMethod("GET"),
    setHeaders(headers)
  );
};
var updateProfile = (data) => request_default(
  setURL(`${API_ROOT}/profile`),
  setMethod("PUT"),
  setData(data, updateProfileSchema)
);
var listGrants = () => request_default(setURL(`${API_ROOT}/profile/grants`));
var getMyGrants = () => request_default(setURL(`${API_ROOT}/profile/grants`), setMethod("GET"));
var getTrustedDevices = (params, filter) => request_default(
  setURL(`${API_ROOT}/profile/devices`),
  setMethod("GET"),
  setXFilter(filter),
  setParams(params)
);
var deleteTrustedDevice = (id) => request_default(
  setURL(`${API_ROOT}/profile/devices/${encodeURIComponent(id)}`),
  setMethod("DELETE")
);
var getUserPreferences = () => {
  return request_default(
    setURL(`${API_ROOT}/profile/preferences`)
  );
};
var updateUserPreferences = (payload) => {
  return request_default(
    setURL(`${API_ROOT}/profile/preferences`),
    setData(payload),
    setMethod("PUT")
  );
};
var getLogins = (params, filter) => {
  return request_default(
    setURL(`${API_ROOT}/profile/logins`),
    setMethod("GET"),
    setXFilter(filter),
    setParams(params)
  );
};
var getSecurityQuestions = () => {
  return request_default(
    setURL(`${API_ROOT}/profile/security-questions`),
    setMethod("GET")
  );
};
var updateSecurityQuestions = (payload) => {
  return request_default(
    setURL(`${API_ROOT}/profile/security-questions`),
    setMethod("POST"),
    setData(payload, SecurityQuestionsSchema)
  );
};
var smsOptOut = () => {
  return request_default(
    setURL(`${API_ROOT}/profile/phone-number`),
    setMethod("DELETE")
  );
};
var sendCodeToPhoneNumber = (data) => {
  return request_default(
    setURL(`${API_ROOT}/profile/phone-number`),
    setMethod("POST"),
    setData(data, SendCodeToPhoneNumberSchema)
  );
};
var verifyPhoneNumberCode = (data) => {
  return request_default(
    setURL(`${API_ROOT}/profile/phone-number/verify`),
    setMethod("POST"),
    setData(data, VerifyPhoneNumberCodeSchema)
  );
};

// src/profile/appTokens.ts
var getAppTokens = (params, filters) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filters),
  setURL(`${API_ROOT}/profile/apps`)
);
var getAppToken = (tokenId) => request_default(
  setMethod("GET"),
  setURL(`${API_ROOT}/profile/apps/${encodeURIComponent(tokenId)}`)
);
var deleteAppToken = (tokenId) => request_default(
  setURL(`${API_ROOT}/profile/apps/${encodeURIComponent(tokenId)}`),
  setMethod("DELETE")
);

// src/profile/accessTokens.ts
import { createPersonalAccessTokenSchema } from "@linode/validation/lib/profile.schema";
var getPersonalAccessTokens = (params, filters) => request_default(
  setMethod("GET"),
  setParams(params),
  setXFilter(filters),
  setURL(`${API_ROOT}/profile/tokens`)
);
var getPersonalAccessToken = (id) => request_default(
  setMethod("GET"),
  setURL(`${API_ROOT}/profile/tokens/${encodeURIComponent(id)}`)
);
var createPersonalAccessToken = (data) => request_default(
  setMethod("POST"),
  setURL(`${API_ROOT}/profile/tokens`),
  setData(data, createPersonalAccessTokenSchema)
);
var updatePersonalAccessToken = (tokenId, data) => request_default(
  setURL(`${API_ROOT}/profile/tokens/${encodeURIComponent(tokenId)}`),
  setMethod("PUT"),
  setData(data, createPersonalAccessTokenSchema)
);
var deletePersonalAccessToken = (tokenId) => request_default(
  setURL(`${API_ROOT}/profile/tokens/${encodeURIComponent(tokenId)}`),
  setMethod("DELETE")
);

// src/regions/constants.ts
var CONTINENT_CODE_TO_CONTINENT = Object.freeze({
  AF: "Africa",
  AN: "Antartica",
  AS: "Asia",
  EU: "Europe",
  NA: "North America",
  OC: "Oceania",
  SA: "South America"
});
var COUNTRY_CODE_TO_CONTINENT_CODE = Object.freeze({
  AD: "EU",
  AE: "AS",
  AF: "AS",
  AG: "NA",
  AI: "NA",
  AL: "EU",
  AM: "AS",
  AO: "AF",
  AQ: "AN",
  AR: "SA",
  AS: "OC",
  AT: "EU",
  AU: "OC",
  AW: "NA",
  AX: "EU",
  AZ: "AS",
  BA: "EU",
  BB: "NA",
  BD: "AS",
  BE: "EU",
  BF: "AF",
  BG: "EU",
  BH: "AS",
  BI: "AF",
  BJ: "AF",
  BL: "NA",
  BM: "NA",
  BN: "AS",
  BO: "SA",
  BQ: "NA",
  BR: "SA",
  BS: "NA",
  BT: "AS",
  BV: "AN",
  BW: "AF",
  BY: "EU",
  BZ: "NA",
  CA: "NA",
  CC: "AS",
  CD: "AF",
  CF: "AF",
  CG: "AF",
  CH: "EU",
  CI: "AF",
  CK: "OC",
  CL: "SA",
  CM: "AF",
  CN: "AS",
  CO: "SA",
  CR: "NA",
  CU: "NA",
  CV: "AF",
  CW: "NA",
  CX: "AS",
  CY: "EU",
  CZ: "EU",
  DE: "EU",
  DJ: "AF",
  DK: "EU",
  DM: "NA",
  DO: "NA",
  DZ: "AF",
  EC: "SA",
  EE: "EU",
  EG: "AF",
  EH: "AF",
  ER: "AF",
  ES: "EU",
  ET: "AF",
  FI: "EU",
  FJ: "OC",
  FK: "SA",
  FM: "OC",
  FO: "EU",
  FR: "EU",
  GA: "AF",
  GB: "EU",
  GD: "NA",
  GE: "AS",
  GF: "SA",
  GG: "EU",
  GH: "AF",
  GI: "EU",
  GL: "NA",
  GM: "AF",
  GN: "AF",
  GP: "NA",
  GQ: "AF",
  GR: "EU",
  GS: "AN",
  GT: "NA",
  GU: "OC",
  GW: "AF",
  GY: "SA",
  HK: "AS",
  HM: "AN",
  HN: "NA",
  HR: "EU",
  HT: "NA",
  HU: "EU",
  ID: "AS",
  IE: "EU",
  IL: "AS",
  IM: "EU",
  IN: "AS",
  IO: "AS",
  IQ: "AS",
  IR: "AS",
  IS: "EU",
  IT: "EU",
  JE: "EU",
  JM: "NA",
  JO: "AS",
  JP: "AS",
  KE: "AF",
  KG: "AS",
  KH: "AS",
  KI: "OC",
  KM: "AF",
  KN: "NA",
  KP: "AS",
  KR: "AS",
  KW: "AS",
  KY: "NA",
  KZ: "AS",
  LA: "AS",
  LB: "AS",
  LC: "NA",
  LI: "EU",
  LK: "AS",
  LR: "AF",
  LS: "AF",
  LT: "EU",
  LU: "EU",
  LV: "EU",
  LY: "AF",
  MA: "AF",
  MC: "EU",
  MD: "EU",
  ME: "EU",
  MF: "NA",
  MG: "AF",
  MH: "OC",
  MK: "EU",
  ML: "AF",
  MM: "AS",
  MN: "AS",
  MO: "AS",
  MP: "OC",
  MQ: "NA",
  MR: "AF",
  MS: "NA",
  MT: "EU",
  MU: "AF",
  MV: "AS",
  MW: "AF",
  MX: "NA",
  MY: "AS",
  MZ: "AF",
  NA: "AF",
  NC: "OC",
  NE: "AF",
  NF: "OC",
  NG: "AF",
  NI: "NA",
  NL: "EU",
  NO: "EU",
  NP: "AS",
  NR: "OC",
  NU: "OC",
  NZ: "OC",
  OM: "AS",
  PA: "NA",
  PE: "SA",
  PF: "OC",
  PG: "OC",
  PH: "AS",
  PK: "AS",
  PL: "EU",
  PM: "NA",
  PN: "OC",
  PR: "NA",
  PS: "AS",
  PT: "EU",
  PW: "OC",
  PY: "SA",
  QA: "AS",
  RE: "AF",
  RO: "EU",
  RS: "EU",
  RU: "EU",
  RW: "AF",
  SA: "AS",
  SB: "OC",
  SC: "AF",
  SD: "AF",
  SE: "EU",
  SG: "AS",
  SH: "AF",
  SI: "EU",
  SJ: "EU",
  SK: "EU",
  SL: "AF",
  SM: "EU",
  SN: "AF",
  SO: "AF",
  SR: "SA",
  SS: "AF",
  ST: "AF",
  SV: "NA",
  SX: "NA",
  SY: "AS",
  SZ: "AF",
  TC: "NA",
  TD: "AF",
  TF: "AN",
  TG: "AF",
  TH: "AS",
  TJ: "AS",
  TK: "OC",
  TL: "OC",
  TM: "AS",
  TN: "AF",
  TO: "OC",
  TR: "AS",
  TT: "NA",
  TV: "OC",
  TW: "AS",
  TZ: "AF",
  UA: "EU",
  UG: "AF",
  UK: "EU",
  // for compatibility with the Linode API
  UM: "OC",
  US: "NA",
  UY: "SA",
  UZ: "AS",
  VA: "EU",
  VC: "NA",
  VE: "SA",
  VG: "NA",
  VI: "NA",
  VN: "AS",
  VU: "OC",
  WF: "OC",
  WS: "OC",
  XK: "EU",
  YE: "AS",
  YT: "AF",
  ZA: "AF",
  ZM: "AF",
  ZW: "AF"
});

// src/regions/regions.ts
var getRegions = (params) => request_default(
  setURL(`${API_ROOT}/regions`),
  setMethod("GET"),
  setParams(params)
);
var getRegion = (regionId) => request_default(
  setURL(`${API_ROOT}/regions/${encodeURIComponent(regionId)}`),
  setMethod("GET")
);
var getRegionAvailabilities = (params, filter) => request_default(
  setURL(`${API_ROOT}/regions/availability`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getRegionAvailability = (regionId) => request_default(
  setURL(`${API_ROOT}/regions/${encodeURIComponent(regionId)}/availability`),
  setMethod("GET")
);

// src/stackscripts/stackscripts.ts
import {
  stackScriptSchema,
  updateStackScriptSchema
} from "@linode/validation/lib/stackscripts.schema";
var getStackScripts = (params, filter) => request_default(
  setURL(`${API_ROOT}/linode/stackscripts`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getStackScript = (stackscriptId) => request_default(
  setURL(
    `${API_ROOT}/linode/stackscripts/${encodeURIComponent(stackscriptId)}`
  ),
  setMethod("GET")
);
var createStackScript = (payload) => request_default(
  setURL(`${API_ROOT}/linode/stackscripts`),
  setMethod("POST"),
  setData(payload, stackScriptSchema)
);
var updateStackScript = (stackscriptId, payload) => request_default(
  setURL(
    `${API_ROOT}/linode/stackscripts/${encodeURIComponent(stackscriptId)}`
  ),
  setMethod("PUT"),
  setData(payload, updateStackScriptSchema)
);
var deleteStackScript = (stackscriptId) => request_default(
  setURL(
    `${API_ROOT}/linode/stackscripts/${encodeURIComponent(stackscriptId)}`
  ),
  setMethod("DELETE")
);

// src/support/support.ts
import {
  createReplySchema,
  createSupportTicketSchema
} from "@linode/validation/lib/support.schema";
var getTickets = (params, filter) => request_default(
  setURL(`${API_ROOT}/support/tickets`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getTicket = (ticketID) => request_default(
  setURL(`${API_ROOT}/support/tickets/${encodeURIComponent(ticketID)}`),
  setMethod("GET")
);
var getTicketReplies = (ticketId, params, filter) => request_default(
  setURL(
    `${API_ROOT}/support/tickets/${encodeURIComponent(ticketId)}/replies`
  ),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var createSupportTicket = (data) => request_default(
  setURL(`${API_ROOT}/support/tickets`),
  setMethod("POST"),
  setData(data, createSupportTicketSchema)
);
var closeSupportTicket = (ticketId) => request_default(
  setURL(`${API_ROOT}/support/tickets/${encodeURIComponent(ticketId)}/close`),
  setMethod("POST")
);
var createReply = (data) => request_default(
  setURL(
    `${API_ROOT}/support/tickets/${encodeURIComponent(
      data.ticket_id
    )}/replies`
  ),
  setMethod("POST"),
  setData(data, createReplySchema)
);
var uploadAttachment = (ticketId, formData) => request_default(
  setURL(
    `${API_ROOT}/support/tickets/${encodeURIComponent(ticketId)}/attachments`
  ),
  setMethod("POST"),
  setData(formData)
);

// src/tags/tags.ts
var getTags = (params, filter) => request_default(
  setURL(`${API_ROOT}/tags`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var createTag = (data) => request_default(setURL(`${API_ROOT}/tags`), setMethod("POST"), setData(data));
var deleteTag = (label) => request_default(
  setURL(`${API_ROOT}/tags/${encodeURIComponent(label)}`),
  setMethod("DELETE")
);

// src/vlans/vlans.ts
var getVlans = (params, filters) => request_default(
  setURL(`${BETA_API_ROOT}/networking/vlans`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filters)
);
var getVlan = (vlanID) => request_default(
  setURL(`${BETA_API_ROOT}/networking/vlans/${encodeURIComponent(vlanID)}`),
  setMethod("GET")
);

// src/volumes/volumes.ts
import {
  CloneVolumeSchema,
  CreateVolumeSchema,
  ResizeVolumeSchema,
  UpdateVolumeSchema
} from "@linode/validation/lib/volumes.schema";
var getVolume = (volumeId) => request_default(
  setURL(`${API_ROOT}/volumes/${encodeURIComponent(volumeId)}`),
  setMethod("GET")
);
var getVolumes = (params, filters) => request_default(
  setURL(`${API_ROOT}/volumes`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filters)
);
var getVolumeTypes = (params) => request_default(
  setURL(`${API_ROOT}/volumes/types`),
  setMethod("GET"),
  setParams(params)
);
var attachVolume = (volumeId, payload) => request_default(
  setURL(`${API_ROOT}/volumes/${encodeURIComponent(volumeId)}/attach`),
  setMethod("POST"),
  setData(payload)
);
var detachVolume = (volumeId) => request_default(
  setURL(`${API_ROOT}/volumes/${encodeURIComponent(volumeId)}/detach`),
  setMethod("POST")
);
var deleteVolume = (volumeId) => request_default(
  setURL(`${API_ROOT}/volumes/${encodeURIComponent(volumeId)}`),
  setMethod("DELETE")
);
var cloneVolume = (volumeId, data) => request_default(
  setURL(`${API_ROOT}/volumes/${encodeURIComponent(volumeId)}/clone`),
  setMethod("POST"),
  setData(data, CloneVolumeSchema)
);
var resizeVolume = (volumeId, data) => request_default(
  setURL(`${API_ROOT}/volumes/${encodeURIComponent(volumeId)}/resize`),
  setMethod("POST"),
  /**
   * Unless we require the old size, we wont be able to validate. We know 10 is the
   * absolute min so it's safe to set here.
   */
  setData(data, ResizeVolumeSchema(10))
);
var updateVolume = (volumeId, data) => request_default(
  setURL(`${API_ROOT}/volumes/${encodeURIComponent(volumeId)}`),
  setMethod("PUT"),
  setData(data, UpdateVolumeSchema)
);
var createVolume = (data) => request_default(
  setURL(`${API_ROOT}/volumes`),
  setMethod("POST"),
  setData(data, CreateVolumeSchema)
);

// src/volumes/migrations.ts
var getVolumesMigrationQueue = (region) => request_default(
  setURL(
    `${BETA_API_ROOT}/regions/${encodeURIComponent(region)}/migration-queue`
  ),
  setMethod("GET")
);
var migrateVolumes = (volumes) => {
  return request_default(
    setURL(`${BETA_API_ROOT}/volumes/migrate`),
    setMethod("POST"),
    setData({ volumes })
  );
};

// src/vpcs/vpcs.ts
import {
  createSubnetSchema,
  createVPCSchema,
  modifySubnetSchema,
  updateVPCSchema
} from "@linode/validation/lib/vpcs.schema";
var getVPCs = (params, filter) => request_default(
  setURL(`${BETA_API_ROOT}/vpcs`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getVPC = (vpcID) => request_default(
  setURL(`${BETA_API_ROOT}/vpcs/${encodeURIComponent(vpcID)}`),
  setMethod("GET")
);
var createVPC = (data) => request_default(
  setURL(`${BETA_API_ROOT}/vpcs`),
  setMethod("POST"),
  setData(data, createVPCSchema)
);
var updateVPC = (vpcID, data) => request_default(
  setURL(`${BETA_API_ROOT}/vpcs/${encodeURIComponent(vpcID)}`),
  setMethod("PUT"),
  setData(data, updateVPCSchema)
);
var deleteVPC = (vpcID) => request_default(
  setURL(`${BETA_API_ROOT}/vpcs/${encodeURIComponent(vpcID)}`),
  setMethod("DELETE")
);
var getSubnets = (vpcID, params, filter) => request_default(
  setURL(`${BETA_API_ROOT}/vpcs/${encodeURIComponent(vpcID)}/subnets`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getSubnet = (vpcID, subnetID) => request_default(
  setURL(
    `${BETA_API_ROOT}/vpcs/${encodeURIComponent(
      vpcID
    )}/subnets/${encodeURIComponent(subnetID)}`
  ),
  setMethod("GET")
);
var createSubnet = (vpcID, data) => request_default(
  setURL(`${BETA_API_ROOT}/vpcs/${encodeURIComponent(vpcID)}/subnets`),
  setMethod("POST"),
  setData(data, createSubnetSchema)
);
var modifySubnet = (vpcID, subnetID, data) => request_default(
  setURL(
    `${BETA_API_ROOT}/vpcs/${encodeURIComponent(
      vpcID
    )}/subnets/${encodeURIComponent(subnetID)}`
  ),
  setMethod("PUT"),
  setData(data, modifySubnetSchema)
);
var deleteSubnet = (vpcID, subnetID) => request_default(
  setURL(
    `${BETA_API_ROOT}/vpcs/${encodeURIComponent(
      vpcID
    )}/subnets/${encodeURIComponent(subnetID)}`
  ),
  setMethod("DELETE")
);

// src/betas/betas.ts
var getBetas = (params, filter) => request_default(
  setURL(`${API_ROOT}/betas`),
  setMethod("GET"),
  setParams(params),
  setXFilter(filter)
);
var getBeta = (betaId) => request_default(
  setURL(`${API_ROOT}/betas/${encodeURIComponent(betaId)}`),
  setMethod("GET")
);
export {
  AFFINITY_TYPES,
  CONTINENT_CODE_TO_CONTINENT,
  COUNTRY_CODE_TO_CONTINENT_CODE,
  acceptEntityTransfer,
  addFirewallDevice,
  addPaymentMethod,
  addPromotion,
  allocateIPAddress,
  allocateIp,
  appendConfigInterface,
  assignAddresses,
  assignLinodesToPlacementGroup,
  attachVolume,
  baseRequest,
  cancelAccount,
  cancelBackups,
  cancelObjectStorage,
  cancelTransfer,
  changeLinodeDiskPassword,
  changeLinodePassword,
  cloneDomain,
  cloneLinode,
  cloneLinodeDisk,
  cloneVolume,
  closeSupportTicket,
  confirmTwoFactor,
  convertYupToLinodeErrors,
  createBasicLoadbalancer,
  createBucket,
  createChildAccountPersonalAccessToken,
  createContact,
  createCredential,
  createDatabase,
  createDomain,
  createDomainRecord,
  createEntityTransfer,
  createFirewall,
  createIPv6Range,
  createImage,
  createKubernetesCluster,
  createLinode,
  createLinodeConfig,
  createLinodeDisk,
  createLoadbalancer,
  createLoadbalancerCertificate,
  createLoadbalancerConfiguration,
  createLoadbalancerRoute,
  createLoadbalancerServiceTarget,
  createLongviewClient,
  createNodeBalancer,
  createNodeBalancerConfig,
  createNodeBalancerConfigNode,
  createNodePool,
  createOAuthClient,
  createObjectStorageKeys,
  createPersonalAccessToken,
  createPlacementGroup,
  createReply,
  createSSHKey,
  createServiceMonitor,
  createStackScript,
  createSubnet,
  createSupportTicket,
  createTag,
  createUser,
  createVPC,
  createVolume,
  deleteAppToken,
  deleteBucket,
  deleteBucketWithRegion,
  deleteContact,
  deleteCredential,
  deleteDatabase,
  deleteDomain,
  deleteDomainRecord,
  deleteFirewall,
  deleteFirewallDevice,
  deleteImage,
  deleteKubernetesCluster,
  deleteLinode,
  deleteLinodeConfig,
  deleteLinodeConfigInterface,
  deleteLinodeDisk,
  deleteLoadbalancer,
  deleteLoadbalancerCertificate,
  deleteLoadbalancerConfiguration,
  deleteLoadbalancerRoute,
  deleteLoadbalancerServiceTarget,
  deleteLongviewClient,
  deleteNodeBalancer,
  deleteNodeBalancerConfig,
  deleteNodeBalancerConfigNode,
  deleteNodePool,
  deleteOAuthClient,
  deletePaymentMethod,
  deletePersonalAccessToken,
  deletePlacementGroup,
  deleteSSHKey,
  deleteSSLCert,
  deleteServiceMonitor,
  deleteStackScript,
  deleteSubnet,
  deleteTag,
  deleteTrustedDevice,
  deleteUser,
  deleteVPC,
  deleteVolume,
  detachVolume,
  disableFirewall,
  disableServiceMonitor,
  disableTwoFactor,
  enableBackups,
  enableFirewall,
  enableManaged,
  enableServiceMonitor,
  enrollInBeta,
  getAccountAgreements,
  getAccountAvailabilities,
  getAccountAvailability,
  getAccountBeta,
  getAccountBetas,
  getAccountInfo,
  getAccountLogin,
  getAccountLogins,
  getAccountMaintenance,
  getAccountSettings,
  getActiveLongviewPlan,
  getAppToken,
  getAppTokens,
  getBeta,
  getBetas,
  getBucket,
  getBucketAccess,
  getBuckets,
  getBucketsInCluster,
  getBucketsInRegion,
  getChildAccount,
  getChildAccounts,
  getClientToken,
  getClusters,
  getConfigInterface,
  getConfigInterfaces,
  getCredentials,
  getDNSZoneFile,
  getDatabaseBackup,
  getDatabaseBackups,
  getDatabaseCredentials,
  getDatabaseEngine,
  getDatabaseEngines,
  getDatabaseType,
  getDatabaseTypes,
  getDatabases,
  getDeprecatedLinodeTypes,
  getDomain,
  getDomainRecord,
  getDomainRecords,
  getDomains,
  getEngineDatabase,
  getEngineDatabases,
  getEntityTransfer,
  getEntityTransfers,
  getEvent,
  getEvents,
  getFirewall,
  getFirewallDevice,
  getFirewallDevices,
  getFirewallRules,
  getFirewalls,
  getGrants,
  getIP,
  getIPs,
  getIPv6Pools,
  getIPv6RangeInfo,
  getIPv6Ranges,
  getImage,
  getImages,
  getInvoice,
  getInvoiceItems,
  getInvoices,
  getKubeConfig,
  getKubernetesCluster,
  getKubernetesClusterDashboard,
  getKubernetesClusterEndpoints,
  getKubernetesClusters,
  getKubernetesVersion,
  getKubernetesVersions,
  getLinode,
  getLinodeBackups,
  getLinodeConfig,
  getLinodeConfigs,
  getLinodeDisk,
  getLinodeDisks,
  getLinodeFirewalls,
  getLinodeIPs,
  getLinodeKernel,
  getLinodeKernels,
  getLinodeLishToken,
  getLinodeSettings,
  getLinodeStats,
  getLinodeStatsByDate,
  getLinodeTransfer,
  getLinodeTransferByDate,
  getLinodeTypes,
  getLinodeVolumes,
  getLinodes,
  getLoadbalancer,
  getLoadbalancerCertificate,
  getLoadbalancerCertificates,
  getLoadbalancerConfiguration,
  getLoadbalancerConfigurations,
  getLoadbalancerConfigurationsEndpointHealth,
  getLoadbalancerEndpointHealth,
  getLoadbalancerRoute,
  getLoadbalancerRoutes,
  getLoadbalancerServiceTargets,
  getLoadbalancers,
  getLogins,
  getLongviewClients,
  getLongviewSubscriptions,
  getManagedContacts,
  getManagedIssues,
  getManagedStats,
  getMyGrants,
  getNetworkUtilization,
  getNodeBalancer,
  getNodeBalancerConfig,
  getNodeBalancerConfigNode,
  getNodeBalancerConfigNodes,
  getNodeBalancerConfigs,
  getNodeBalancerFirewalls,
  getNodeBalancerStats,
  getNodeBalancerTypes,
  getNodeBalancers,
  getNodePool,
  getNodePools,
  getNotifications,
  getOAuthClient,
  getOAuthClients,
  getObjectACL,
  getObjectList,
  getObjectStorageKeys,
  getObjectURL,
  getPaymentMethod,
  getPaymentMethods,
  getPayments,
  getPersonalAccessToken,
  getPersonalAccessTokens,
  getPlacementGroup,
  getPlacementGroups,
  getProfile,
  getRegion,
  getRegionAvailabilities,
  getRegionAvailability,
  getRegions,
  getSSHKey,
  getSSHKeys,
  getSSHPubKey,
  getSSLCert,
  getSSLFields,
  getSecurityQuestions,
  getServiceTarget,
  getServiceTargetsEndpointHealth,
  getServices,
  getStackScript,
  getStackScripts,
  getSubnet,
  getSubnets,
  getTFAToken,
  getTags,
  getTicket,
  getTicketReplies,
  getTickets,
  getTrustedDevices,
  getType,
  getUser,
  getUserPreferences,
  getUsers,
  getVPC,
  getVPCs,
  getVlan,
  getVlans,
  getVolume,
  getVolumeTypes,
  getVolumes,
  getVolumesMigrationQueue,
  importZone,
  isEmpty,
  linodeBoot,
  linodeReboot,
  linodeShutdown,
  listGrants,
  makeDefaultPaymentMethod,
  makePayment,
  markEventRead,
  markEventSeen,
  migrateVolumes,
  modifySubnet,
  rebuildLinode,
  recycleAllNodes,
  recycleClusterNodes,
  recycleNode,
  removeIPAddress,
  removeIPv6Range,
  rescueLinode,
  rescueMetalLinode,
  resetDatabaseCredentials,
  resetKubeConfig,
  resetOAuthClientSecret,
  resizeLinode,
  resizeLinodeDisk,
  resizeVolume,
  restoreBackup,
  restoreWithBackup,
  revokeObjectStorageKey,
  saveCreditCard,
  scheduleOrQueueMigration,
  sendCodeToPhoneNumber,
  setToken,
  shareAddresses,
  shareAddressesv4,
  signAgreement,
  smsOptOut,
  startMutation,
  takeSnapshot,
  unassignLinodesFromPlacementGroup,
  updateAccountInfo,
  updateAccountSettings,
  updateActiveLongviewPlan,
  updateBucketAccess,
  updateConfigInterface,
  updateContact,
  updateCredential,
  updateDatabase,
  updateDomain,
  updateDomainRecord,
  updateFirewall,
  updateFirewallRules,
  updateGrants,
  updateIP,
  updateImage,
  updateKubernetesCluster,
  updateLinode,
  updateLinodeConfig,
  updateLinodeConfigOrder,
  updateLinodeDisk,
  updateLinodeSettings,
  updateLoadbalancer,
  updateLoadbalancerCertificate,
  updateLoadbalancerConfiguration,
  updateLoadbalancerRoute,
  updateLoadbalancerServiceTarget,
  updateLongviewClient,
  updateNodeBalancer,
  updateNodeBalancerConfig,
  updateNodeBalancerConfigNode,
  updateNodePool,
  updateOAuthClient,
  updateObjectACL,
  updateObjectStorageKey,
  updatePassword,
  updatePersonalAccessToken,
  updatePlacementGroup,
  updateProfile,
  updateSSHKey,
  updateSecurityQuestions,
  updateServiceMonitor,
  updateStackScript,
  updateUser,
  updateUserPreferences,
  updateVPC,
  updateVolume,
  uploadAttachment,
  uploadImage,
  uploadSSLCert,
  verifyPhoneNumberCode
};
